<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Michael Liou">
<meta name="dcterms.date" content="2021-02-21">

<title>Personal Documentation - 14&nbsp; Longitudinal Data Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../longitudinal/randomcoef.html" rel="next">
<link href="../gam/gam.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

<script src="../site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script>window.backupDefine = window.define; window.define = undefined;</script><script src="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
 var mathElements = document.getElementsByClassName("math");
 var macros = [];
 for (var i = 0; i < mathElements.length; i++) {
  var texText = mathElements[i].firstChild;
  if (mathElements[i].tagName == "SPAN") {
   katex.render(texText.data, mathElements[i], {
    displayMode: mathElements[i].classList.contains('display'),
    throwOnError: false,
    macros: macros,
    fleqn: false
   });
}}});
  </script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.1/dist/katex.min.css">

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Longitudinal Data Analysis</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Personal Documentation</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Front Page</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../advanced/advanced.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced R</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../approximation/approximation.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Approximation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../bayesian/bayesian.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Bayesian</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../categorical/categorical.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Categorical Data Analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../changepoint/changepoint.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Changepoint Analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chaos/chaos.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Chaos</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../crossover/crossover.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Crossover Designs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../diffeq/diffeq.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Differential Equations</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../epidemiology/epidemiology.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Epidemiology</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../experimentaldesign/experimentaldesign.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Experimental Design</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../exponential/exponential.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Exponential Modeling</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../gam/gam.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Generalized Additive Models</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../longitudinal/longitudinal.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Longitudinal Data Analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../longitudinal/randomcoef.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Random Coefficient Models (CALS)</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../longitudinal/within_between.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">The Within-Between Model</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../matrix/matrix.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Matrix Math</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../meta/meta.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Meta Analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../mixed/mixed.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Mixed Models</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../mixed/mixed_variance.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Mixed Models (Variance)</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../network/network_centrality.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Network Centrality</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../network/network_gnar.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">GNAR</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../next_gen_sequencing/next_gen_sequencing.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Next Generation Sequencing</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../nonlinear/nonlinear.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Nonlinear</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../observable/observable.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Observable Testing</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../optimization/optimization.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Optimization</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../ordination/ordination.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Ordination</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../parallel/parallel.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Parallel Computing in R</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../plant_breeding/plant_breeding.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Plant Breeding</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../social_network/social_network.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">30</span>&nbsp; <span class="chapter-title">Social Network Analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../spatial/carbayes.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">31</span>&nbsp; <span class="chapter-title">CARBayes</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../spatial/spatial.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">32</span>&nbsp; <span class="chapter-title">Spatial Statistics</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../stochastic/stochastic.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">33</span>&nbsp; <span class="chapter-title">Stochastic</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../stochastic/ctmc.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">34</span>&nbsp; <span class="chapter-title">CTMC</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../timeseries/timeseries.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">35</span>&nbsp; <span class="chapter-title">Time Series</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Draft</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../bootstrap/bootstrap.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">36</span>&nbsp; <span class="chapter-title">Bootstrap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../causal/causal.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">37</span>&nbsp; <span class="chapter-title">Causal Statistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../sensitivity/sensitivity.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">38</span>&nbsp; <span class="chapter-title">Sensitivity Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../signal/signal.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">39</span>&nbsp; <span class="chapter-title">Signal Processing</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Appendices</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../color/color.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Colors in R</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#why-longitudinal-data-analysis" id="toc-why-longitudinal-data-analysis" class="nav-link active" data-scroll-target="#why-longitudinal-data-analysis"><span class="toc-section-number">15</span>  Why Longitudinal Data Analysis</a>
  <ul class="collapse">
  <li><a href="#specifics-of-lda" id="toc-specifics-of-lda" class="nav-link" data-scroll-target="#specifics-of-lda"><span class="toc-section-number">15.1</span>  Specifics of LDA</a></li>
  <li><a href="#approaches-in-lda" id="toc-approaches-in-lda" class="nav-link" data-scroll-target="#approaches-in-lda"><span class="toc-section-number">15.2</span>  Approaches in LDA</a></li>
  <li><a href="#historical-methods" id="toc-historical-methods" class="nav-link" data-scroll-target="#historical-methods"><span class="toc-section-number">15.3</span>  Historical Methods</a>
  <ul class="collapse">
  <li><a href="#split-plot-in-time" id="toc-split-plot-in-time" class="nav-link" data-scroll-target="#split-plot-in-time"><span class="toc-section-number">15.3.1</span>  Split plot in time</a></li>
  <li><a href="#repeated-measures-anova" id="toc-repeated-measures-anova" class="nav-link" data-scroll-target="#repeated-measures-anova"><span class="toc-section-number">15.3.2</span>  Repeated Measures ANOVA</a></li>
  <li><a href="#manova" id="toc-manova" class="nav-link" data-scroll-target="#manova"><span class="toc-section-number">15.3.3</span>  MANOVA</a></li>
  <li><a href="#summary-values" id="toc-summary-values" class="nav-link" data-scroll-target="#summary-values"><span class="toc-section-number">15.3.4</span>  Summary values</a></li>
  </ul></li>
  <li><a href="#inference-on-parameters" id="toc-inference-on-parameters" class="nav-link" data-scroll-target="#inference-on-parameters"><span class="toc-section-number">15.4</span>  Inference on parameters</a></li>
  </ul></li>
  <li><a href="#the-setup" id="toc-the-setup" class="nav-link" data-scroll-target="#the-setup"><span class="toc-section-number">16</span>  The setup</a></li>
  <li><a href="#modeling-the-mean" id="toc-modeling-the-mean" class="nav-link" data-scroll-target="#modeling-the-mean"><span class="toc-section-number">17</span>  Modeling the Mean</a>
  <ul class="collapse">
  <li><a href="#example-tlc" id="toc-example-tlc" class="nav-link" data-scroll-target="#example-tlc"><span class="toc-section-number">17.1</span>  Example TLC</a>
  <ul class="collapse">
  <li><a href="#sas" id="toc-sas" class="nav-link" data-scroll-target="#sas"><span class="toc-section-number">17.1.1</span>  SAS</a></li>
  <li><a href="#r-gls" id="toc-r-gls" class="nav-link" data-scroll-target="#r-gls"><span class="toc-section-number">17.1.2</span>  R: gls</a></li>
  <li><a href="#r-anova.gls" id="toc-r-anova.gls" class="nav-link" data-scroll-target="#r-anova.gls"><span class="toc-section-number">17.1.3</span>  R: anova.gls</a></li>
  <li><a href="#r-lm" id="toc-r-lm" class="nav-link" data-scroll-target="#r-lm"><span class="toc-section-number">17.1.4</span>  R: lm</a></li>
  <li><a href="#r-auc" id="toc-r-auc" class="nav-link" data-scroll-target="#r-auc"><span class="toc-section-number">17.1.5</span>  R: AUC</a></li>
  </ul></li>
  <li><a href="#example-ratdrink" id="toc-example-ratdrink" class="nav-link" data-scroll-target="#example-ratdrink"><span class="toc-section-number">17.2</span>  Example: Ratdrink</a>
  <ul class="collapse">
  <li><a href="#different-intercepts" id="toc-different-intercepts" class="nav-link" data-scroll-target="#different-intercepts"><span class="toc-section-number">17.2.1</span>  different intercepts</a></li>
  <li><a href="#different-slopes" id="toc-different-slopes" class="nav-link" data-scroll-target="#different-slopes"><span class="toc-section-number">17.2.2</span>  different slopes</a></li>
  </ul></li>
  <li><a href="#example-body-fat-and-menarche" id="toc-example-body-fat-and-menarche" class="nav-link" data-scroll-target="#example-body-fat-and-menarche"><span class="toc-section-number">17.3</span>  Example: Body Fat and Menarche</a>
  <ul class="collapse">
  <li><a href="#modeling" id="toc-modeling" class="nav-link" data-scroll-target="#modeling"><span class="toc-section-number">17.3.1</span>  Modeling</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#testing-for-heteroskedasticiy-and-autocorrelation" id="toc-testing-for-heteroskedasticiy-and-autocorrelation" class="nav-link" data-scroll-target="#testing-for-heteroskedasticiy-and-autocorrelation"><span class="toc-section-number">18</span>  Testing for heteroskedasticiy and autocorrelation</a>
  <ul class="collapse">
  <li><a href="#resources" id="toc-resources" class="nav-link" data-scroll-target="#resources"><span class="toc-section-number">18.1</span>  Resources</a></li>
  </ul></li>
  <li><a href="#modeling-the-covariance" id="toc-modeling-the-covariance" class="nav-link" data-scroll-target="#modeling-the-covariance"><span class="toc-section-number">19</span>  Modeling the Covariance</a>
  <ul class="collapse">
  <li><a href="#examples" id="toc-examples" class="nav-link" data-scroll-target="#examples"><span class="toc-section-number">19.1</span>  Examples</a>
  <ul class="collapse">
  <li><a href="#example-spruce-trees" id="toc-example-spruce-trees" class="nav-link" data-scroll-target="#example-spruce-trees"><span class="toc-section-number">19.1.1</span>  Example: Spruce Trees</a></li>
  <li><a href="#exploring-correlation-from-week-to-week-and-getting-correlation-structure." id="toc-exploring-correlation-from-week-to-week-and-getting-correlation-structure." class="nav-link" data-scroll-target="#exploring-correlation-from-week-to-week-and-getting-correlation-structure."><span class="toc-section-number">19.1.2</span>  Exploring correlation from week to week, and getting correlation structure.</a></li>
  <li><a href="#example-tlc-1" id="toc-example-tlc-1" class="nav-link" data-scroll-target="#example-tlc-1"><span class="toc-section-number">19.1.3</span>  Example: TLC</a></li>
  <li><a href="#example-body-fat-and-menarche-1" id="toc-example-body-fat-and-menarche-1" class="nav-link" data-scroll-target="#example-body-fat-and-menarche-1"><span class="toc-section-number">19.1.4</span>  Example: Body Fat and Menarche</a></li>
  <li><a href="#example-dental" id="toc-example-dental" class="nav-link" data-scroll-target="#example-dental"><span class="toc-section-number">19.1.5</span>  Example: Dental</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#random-coefficient-models" id="toc-random-coefficient-models" class="nav-link" data-scroll-target="#random-coefficient-models"><span class="toc-section-number">20</span>  Random Coefficient Models</a>
  <ul class="collapse">
  <li><a href="#example-autism" id="toc-example-autism" class="nav-link" data-scroll-target="#example-autism"><span class="toc-section-number">20.1</span>  Example: Autism</a></li>
  </ul></li>
  <li><a href="#baseline-adjustment" id="toc-baseline-adjustment" class="nav-link" data-scroll-target="#baseline-adjustment"><span class="toc-section-number">21</span>  Baseline adjustment</a>
  <ul class="collapse">
  <li><a href="#method-implementations" id="toc-method-implementations" class="nav-link" data-scroll-target="#method-implementations"><span class="toc-section-number">21.1</span>  Method Implementations</a>
  <ul class="collapse">
  <li><a href="#raw" id="toc-raw" class="nav-link" data-scroll-target="#raw"><span class="toc-section-number">21.1.1</span>  1. raw</a></li>
  <li><a href="#constrain-baseline" id="toc-constrain-baseline" class="nav-link" data-scroll-target="#constrain-baseline"><span class="toc-section-number">21.1.2</span>  2. constrain baseline</a></li>
  <li><a href="#change-from-baseline" id="toc-change-from-baseline" class="nav-link" data-scroll-target="#change-from-baseline"><span class="toc-section-number">21.1.3</span>  3. change from baseline</a></li>
  <li><a href="#ancova-change" id="toc-ancova-change" class="nav-link" data-scroll-target="#ancova-change"><span class="toc-section-number">21.1.4</span>  4. ancova change</a></li>
  </ul></li>
  <li><a href="#comparison" id="toc-comparison" class="nav-link" data-scroll-target="#comparison"><span class="toc-section-number">21.2</span>  Comparison</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="toc-section-number">21.3</span>  References</a></li>
  </ul></li>
  <li><a href="#residuals" id="toc-residuals" class="nav-link" data-scroll-target="#residuals"><span class="toc-section-number">22</span>  Residuals</a>
  <ul class="collapse">
  <li><a href="#decorrelation" id="toc-decorrelation" class="nav-link" data-scroll-target="#decorrelation"><span class="toc-section-number">22.1</span>  Decorrelation</a></li>
  <li><a href="#aggregation" id="toc-aggregation" class="nav-link" data-scroll-target="#aggregation"><span class="toc-section-number">22.2</span>  Aggregation</a></li>
  </ul></li>
  <li><a href="#econometrics-approaches" id="toc-econometrics-approaches" class="nav-link" data-scroll-target="#econometrics-approaches"><span class="toc-section-number">23</span>  Econometrics Approaches</a></li>
  <li><a href="#software-comparisons" id="toc-software-comparisons" class="nav-link" data-scroll-target="#software-comparisons"><span class="toc-section-number">24</span>  Software Comparisons</a></li>
  <li><a href="#plm-vignette" id="toc-plm-vignette" class="nav-link" data-scroll-target="#plm-vignette"><span class="toc-section-number">25</span>  PLM Vignette</a>
  <ul class="collapse">
  <li><a href="#plm" id="toc-plm" class="nav-link" data-scroll-target="#plm"><span class="toc-section-number">25.1</span>  <code>plm</code></a>
  <ul class="collapse">
  <li><a href="#empluk" id="toc-empluk" class="nav-link" data-scroll-target="#empluk"><span class="toc-section-number">25.1.1</span>  EmplUK</a></li>
  <li><a href="#grunfield-example" id="toc-grunfield-example" class="nav-link" data-scroll-target="#grunfield-example"><span class="toc-section-number">25.1.2</span>  Grunfield example</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#gls" id="toc-gls" class="nav-link" data-scroll-target="#gls"><span class="toc-section-number">26</span>  GLS</a></li>
  <li><a href="#References" id="toc-References" class="nav-link" data-scroll-target="#References"><span class="toc-section-number">27</span>  References</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Longitudinal Data Analysis</span></h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Michael Liou </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 21, 2021</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-1_7b3e4fcf70130e30e557f2cf41430e5a">
<details>
<summary>Libraries</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plm)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(skimr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sloop) <span class="co"># For object sloothing</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lme4) <span class="co"># mixed</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(emmeans)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(faraway)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(foreign)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lmeInfo) <span class="co"># for information matrices of lme models</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(broom.mixed)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ProfileLikelihood)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sandwich)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lmtest)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="why-longitudinal-data-analysis" class="level1" data-number="15">
<h1 data-number="15"><span class="header-section-number">15</span> Why Longitudinal Data Analysis</h1>
<p>More information about the change, at the trade-off of needing to take repeated measurments.</p>
<section id="specifics-of-lda" class="level2" data-number="15.1">
<h2 data-number="15.1" class="anchored" data-anchor-id="specifics-of-lda"><span class="header-section-number">15.1</span> Specifics of LDA</h2>
<ul>
<li>Replication is a series of observations (each subject) and not individual measurements. Thus the EU is the subject.</li>
</ul>
</section>
<section id="approaches-in-lda" class="level2" data-number="15.2">
<h2 data-number="15.2" class="anchored" data-anchor-id="approaches-in-lda"><span class="header-section-number">15.2</span> Approaches in LDA</h2>
<ul>
<li>Marginal (averages over population. state employee perspective)</li>
<li>(Random effects) Mixed Models (doctor perspective, subject specifics)</li>
<li>Transition Model (Ware et al, 1988)</li>
</ul>
</section>
<section id="historical-methods" class="level2" data-number="15.3">
<h2 data-number="15.3" class="anchored" data-anchor-id="historical-methods"><span class="header-section-number">15.3</span> Historical Methods</h2>
<section id="split-plot-in-time" class="level3" data-number="15.3.1">
<h3 data-number="15.3.1" class="anchored" data-anchor-id="split-plot-in-time"><span class="header-section-number">15.3.1</span> Split plot in time</h3>
<ul>
<li>Induces a compound symmetric structure with observations</li>
</ul>
</section>
<section id="repeated-measures-anova" class="level3" data-number="15.3.2">
<h3 data-number="15.3.2" class="anchored" data-anchor-id="repeated-measures-anova"><span class="header-section-number">15.3.2</span> Repeated Measures ANOVA</h3>
<p>I think this is the same as split plot in time… The model is</p>
<p><span class="math display">
Y_{ij} = X_{ij}'\beta + b_i + e_{ij}
</span> The covariance structure is <em>compound symmetry</em>, meaning</p>
<p><span class="math display">
\begin{aligned}
\text{Cov}(Y_{i}) =
\begin{bmatrix}
\sigma_b + \sigma_e &amp; \sigma_e &amp; \dots &amp;\sigma_e \\
\sigma_e &amp; \sigma_b + \sigma_e &amp; \dots &amp;\sigma_e \\
\vdots &amp; \vdots &amp; \ddots &amp; \vdots \\
\sigma_e &amp;  \sigma_e &amp; \dots &amp; \sigma_b + \sigma_e \\
\end{bmatrix}
\end{aligned}
</span></p>
</section>
<section id="manova" class="level3" data-number="15.3.3">
<h3 data-number="15.3.3" class="anchored" data-anchor-id="manova"><span class="header-section-number">15.3.3</span> MANOVA</h3>
<p>Special case of the so called <em>profile analysis</em>. The main idea for MANOVA is to make some trasnformations and make some derived response varaibles to analyze. The multiple comes from having multiple responses. One is to make the sum of all the responses, to examine average over time. You could also create a variable for linear change across time, or quadratic change within subject.</p>
<p>There are some disadvantages though,</p>
<ul>
<li>If design is unbalanced across time, MANOVA can’t be used.</li>
<li>Also if there are any missing data, the entire case must be thrown out.</li>
</ul>
</section>
<section id="summary-values" class="level3" data-number="15.3.4">
<h3 data-number="15.3.4" class="anchored" data-anchor-id="summary-values"><span class="header-section-number">15.3.4</span> Summary values</h3>
<p>Reduce the sequence of each individual to a small set of summary values. Then, you can use the classic t-test or ANOVA, univariate tests.</p>
<ul>
<li>Area under curve (AUC) - Can only be used when the people have the same time measurements.</li>
</ul>
<p>Drawbacks</p>
<ul>
<li>forces data analyst to think about one aspect of the repeated measures.</li>
<li>Might have the same summary measure but different response profile.</li>
<li>Method can’t be applied if one of the covariates is time varying. b/c variance will not be constant from summary to summary.</li>
</ul>
</section>
</section>
<section id="inference-on-parameters" class="level2" data-number="15.4">
<h2 data-number="15.4" class="anchored" data-anchor-id="inference-on-parameters"><span class="header-section-number">15.4</span> Inference on parameters</h2>
<p>Likelihood test requires an additional fit on the null hypothesis, but better properties. Recommend Likelihood ratio based tests and CI. Note the RMLE is a correction for the data estimating both the mean and covariance.</p>
</section>
</section>
<section id="the-setup" class="level1" data-number="16">
<h1 data-number="16"><span class="header-section-number">16</span> The setup</h1>
<p>The short hand notation for the observations of a single individual are:</p>
<p><span class="math display">
\begin{aligned}
Y_{i} = X_{i}\beta + \varepsilon_{i}
\end{aligned}
</span> with <span class="math inline">\varepsilon_i \sim N(0, \Sigma)</span>, which expanded would look like,</p>
<p><span class="math display">
\begin{aligned}
\begin{bmatrix}
y_{i1} \\
y_{i2} \\
\vdots \\
y_{in_{i}} \\
\end{bmatrix} =
\begin{bmatrix}
X_{i11} &amp; X_{i12} &amp; \dots  &amp;X_{i1p}  \\
X_{i21} &amp;  X_{i22} &amp; \dots  &amp;X_{i2p}  \\
\vdots &amp;  \vdots &amp; \ddots  &amp; \vdots  \\
X_{in_{i}1} &amp;  X_{in_{i}2} &amp; \dots  &amp; X_{in_{i}p}  \\
\end{bmatrix}
\begin{bmatrix}
\beta_1 \\
\beta_2 \\
\vdots \\
\beta_p
\end{bmatrix} +
\begin{bmatrix}
\varepsilon_{i1} \\
\varepsilon_{i2} \\
\vdots \\
\varepsilon_{in_{i}}
\end{bmatrix}
\end{aligned}
</span></p>
<p>In total, we would stack them, so we have</p>
<p><span class="math display">
\begin{aligned}
\begin{bmatrix}
Y_1 \\
Y_2 \\
\vdots \\
Y_N \\
\end{bmatrix} =
\begin{bmatrix}
X_1 \\
X_2 \\
\vdots \\
X_N \\
\end{bmatrix}
\begin{bmatrix}
\beta_1 \\
\beta_2 \\
\vdots \\
\beta_p
\end{bmatrix} +
\begin{bmatrix}
\mathbf{\varepsilon_{1}} \\
\mathbf{\varepsilon_{2}} \\
\vdots \\
\mathbf{\varepsilon_{N}}
\end{bmatrix}
\end{aligned}
</span> Thus, <span class="math inline">\varepsilon</span> has a block diagonal structure with each subject having the same covariance matrix.</p>
</section>
<section id="modeling-the-mean" class="level1" data-number="17">
<h1 data-number="17"><span class="header-section-number">17</span> Modeling the Mean</h1>
<p>This section follows 5.5 in <span class="citation" data-cites="fitzmaurice_applied_2011">Fitzmaurice, Laird, and Ware (<a href="#ref-fitzmaurice_applied_2011" role="doc-biblioref">2011</a>)</span>.</p>
<p>The main questions to ask in this model are</p>
<ol type="1">
<li>Group x time interaction effect - do the group means change over time. Biggest question of longitudinal studies normally</li>
<li>Group effect - a main effect, but similar in normal analyses because if the interaction is significant, then, it’s relatively meaningless to talk about these effects. (Note, if randomized trial, this should coincide with the interaction, because in randomized trial it was assumed that the two groups were the same.)</li>
<li>Time effect - are they same across time?</li>
</ol>
<p>There are primarily two ways of studying the longitudinal responses</p>
<ol type="1">
<li>response profiles (perfect fit, treating time as a factor)</li>
<li>(semi-) parametric form</li>
<li>summary measure (like area under the curve)</li>
</ol>
<section id="example-tlc" class="level2 tabset" data-number="17.1">
<h2 class="tabset anchored" data-number="17.1" data-anchor-id="example-tlc"><span class="header-section-number">17.1</span> Example TLC</h2>
<p>This is an exploratory plot summarizing the eventual statements we’d like to say about the model.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-2_38a1dfe48c61fe9dc625d1dba15f25db">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>tlc_raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/tlc.csv"</span>, <span class="at">header=</span><span class="cn">FALSE</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tlc_raw) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"trt"</span>,<span class="st">"w0"</span>, <span class="st">"w1"</span>, <span class="st">"w4"</span>,<span class="st">"w6"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>tlc <span class="ot">&lt;-</span> tlc_raw <span class="sc">%&gt;%</span> <span class="fu">gather</span>(<span class="st">"week"</span>, <span class="st">"lead"</span>, <span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># numeric version of week</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>tlc<span class="sc">$</span>week_int <span class="ot">&lt;-</span> tlc<span class="sc">$</span>week <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">".*([0-9]+).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, .) <span class="sc">%&gt;%</span> <span class="fu">as.numeric</span>()</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co"># timepoint (1:4)</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>tlc<span class="sc">$</span>time <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)[<span class="fu">as.factor</span>(tlc<span class="sc">$</span>week)]</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># change from baseline</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>tlc_diff <span class="ot">&lt;-</span> tlc <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(trt, id, time) <span class="sc">%&gt;%</span> </span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">lead_diff =</span> lead <span class="sc">-</span> lead[<span class="dv">1</span>],</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">lead_base =</span> lead[<span class="dv">1</span>],</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">trt =</span> <span class="fu">factor</span>(trt, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"P"</span>, <span class="st">"A"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(trt), id, time) <span class="sc">%&gt;%</span> </span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(week <span class="sc">!=</span> <span class="st">"w0"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-3_30dc1d1eddffac3eae9fbc7750d47ab1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>tlc <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> week, <span class="at">y =</span> lead, <span class="at">group =</span> id)) <span class="sc">+</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">alpha =</span> .<span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>trt)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-4_871afba696f2806fcbb0343d19a19faf">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean Response Profile by Trt</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>tlc_mrp <span class="ot">&lt;-</span> tlc <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(trt, week) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mlead =</span> <span class="fu">mean</span>(lead), </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">.groups =</span> <span class="st">"drop_last"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>tlc_mrp <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> week, <span class="at">y=</span>mlead, <span class="at">group=</span>trt)) <span class="sc">+</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">linetype=</span>trt)) <span class="sc">+</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Lead over time"</span>) <span class="sc">+</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="sas" class="level3" data-number="17.1.1">
<h3 data-number="17.1.1" class="anchored" data-anchor-id="sas"><span class="header-section-number">17.1.1</span> SAS</h3>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-5_df84558dba335a7b25240588c7212dd7">
<details>
<summary>Code</summary>
<pre class="sas cell-code"><code>* TLC data analysis example, from chapter 5;
* https://content.sph.harvard.edu/fitzmaur/ala2e/;

DATA tlc_long;
  INFILE "~/lda/tlc-data.txt" DLM=" ";
  input id group $ lead0 lead1 lead4 lead6;
  * create 4 observations from each row;
  y=lead0; time=0; output;
  y=lead1; time=1; output;
  y=lead4; time=4; output;
  y=lead6; time=6; output;
  drop lead0 lead1 lead4 lead6; * drop original "wide" data columns;
run;

/* set reference level */
/* http://support.sas.com/kb/37/108.html */
proc mixed noclprint=10 data=tlc_long
class id group(ref="P") time(ref="0");
model y = group time group*time / s chisq;
repeated time / type=un subject=id r;
lsmeans group / cl diff;
run;</code></pre>
</details>
</div>
<center>
<figure class="figure">
<p><img src="img/tlc_sas_summary.png" style="width:50%" class="figure-img"></p>
<figcaption class="figure-caption">
</figcaption>
</figure>
<figure class="figure">
<p><img src="img/tlc_sas_type3.png" style="width:50%" class="figure-img"></p>
<figcaption class="figure-caption">
</figcaption>
</figure>
<figure class="figure">
<p><img src="img/tlc_sas_cov.png" style="width:50%" class="figure-img"></p>
<figcaption class="figure-caption">
</figcaption>
</figure>
</center>
</section>
<section id="r-gls" class="level3" data-number="17.1.2">
<h3 data-number="17.1.2" class="anchored" data-anchor-id="r-gls"><span class="header-section-number">17.1.2</span> R: gls</h3>
<p>We fit an unstructured covariance, with fully crossed time and treatment factors. In order to get the unstructured covariance matrix, we must use <code>gls</code>, for generalized least squares. This is basically</p>
<div class="cell" data-paged.print="true" data-hash="longitudinal_cache/html/modeling_415ebc6c0168d7ee2bf7b7bf3d8a75ba">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>tlc<span class="sc">$</span>trt <span class="ot">&lt;-</span> <span class="fu">factor</span>(tlc<span class="sc">$</span>trt, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"P"</span>, <span class="st">"A"</span>)) <span class="co"># use P as the reference level</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Generalized Least Squares, defaults to REML</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>tlc_gls <span class="ot">&lt;-</span> <span class="fu">gls</span>(lead <span class="sc">~</span> trt<span class="sc">*</span>week,</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">corr=</span><span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> time <span class="sc">|</span> id),</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> week),</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">data=</span>tlc)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Table 5.5 in Fitzmaurice</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(tlc_gls)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Generalized least squares fit by REML
  Model: lead ~ trt * week 
  Data: tlc 
       AIC      BIC    logLik
  2452.076 2523.559 -1208.038

Correlation Structure: General
 Formula: ~time | id 
 Parameter estimate(s):
 Correlation: 
  1     2     3    
2 0.571            
3 0.570 0.775      
4 0.577 0.582 0.581
Variance function:
 Structure: Different standard deviations per stratum
 Formula: ~1 | week 
 Parameter estimates:
      w0       w1       w4       w6 
1.000000 1.325880 1.370442 1.524813 

Coefficients:
              Value Std.Error   t-value p-value
(Intercept)  26.272 0.7102929  36.98756  0.0000
trtA          0.268 1.0045059   0.26680  0.7898
weekw1       -1.612 0.7919199  -2.03556  0.0425
weekw4       -2.202 0.8149021  -2.70217  0.0072
weekw6       -2.626 0.8885253  -2.95546  0.0033
trtA:weekw1 -11.406 1.1199438 -10.18444  0.0000
trtA:weekw4  -8.824 1.1524456  -7.65676  0.0000
trtA:weekw6  -3.152 1.2565645  -2.50843  0.0125

 Correlation: 
            (Intr) trtA   weekw1 weekw4 weekw6 trtA:1 trtA:4
trtA        -0.707                                          
weekw1      -0.218  0.154                                   
weekw4      -0.191  0.135  0.680                            
weekw6      -0.096  0.068  0.386  0.385                     
trtA:weekw1  0.154 -0.218 -0.707 -0.481 -0.273              
trtA:weekw4  0.135 -0.191 -0.481 -0.707 -0.272  0.680       
trtA:weekw6  0.068 -0.096 -0.273 -0.272 -0.707  0.386  0.385

Standardized residuals:
       Min         Q1        Med         Q3        Max 
-2.1756456 -0.6849980 -0.1515547  0.5294176  5.6327571 

Residual standard error: 5.02253 
Degrees of freedom: 400 total; 392 residual</code></pre>
</div>
</div>
<p>The estimated (unstructured), marginal covariance structure can be extracted by <code>getVarCov</code>.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-6_dd765b2111bc5aee2faf8c59545b4cf2">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(tlc_gls) <span class="co"># covariance matrix</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Marginal variance covariance matrix
       [,1]   [,2]   [,3]   [,4]
[1,] 25.226 19.107 19.700 22.202
[2,] 19.107 44.346 35.535 29.675
[3,] 19.700 35.535 47.377 30.620
[4,] 22.202 29.675 30.620 58.651
  Standard Deviations: 5.0225 6.6593 6.8831 7.6584 </code></pre>
</div>
</div>
<p>Getting the type III fixed effect tests, similar to SAS, is a little more work. There must be an easier way, but this shows how to do it manually. Here P0 denotes the mean of placebo at time 0.</p>
<table class="table">
<thead>
<tr class="header">
<th>Conditional Mean (<span class="math inline">\mu</span>)</th>
<th>Coef</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>P0</td>
<td><span class="math inline">\beta_1</span></td>
</tr>
<tr class="even">
<td>P1</td>
<td><span class="math inline">\beta_1 + \beta_3</span></td>
</tr>
<tr class="odd">
<td>P4</td>
<td><span class="math inline">\beta_1 + \beta_4</span></td>
</tr>
<tr class="even">
<td>P6</td>
<td><span class="math inline">\beta_1 + \beta_5</span></td>
</tr>
<tr class="odd">
<td>S0</td>
<td><span class="math inline">\beta_1 + \beta_2</span></td>
</tr>
<tr class="even">
<td>S1</td>
<td><span class="math inline">\beta_1 + \beta_2 + \beta_3 + \beta_6</span></td>
</tr>
<tr class="odd">
<td>S4</td>
<td><span class="math inline">\beta_1 + \beta_2 + \beta_4 + \beta_7</span></td>
</tr>
<tr class="even">
<td>S7</td>
<td><span class="math inline">\beta_1 + \beta_2 + \beta_5 + \beta_8</span></td>
</tr>
</tbody>
</table>
<p>Furthermore, the Wald test statistic is</p>
<p><span class="math display">
\begin{aligned}
W^2 = (L\hat\beta)'\{L\mathrm{Cov}(\hat\beta)L'\}^{-1}L\hat\beta
\end{aligned}
</span> Compare the following table to the SAS Type 3 tests for Fixed Effects</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-7_5a636d3df51a8f73f2cbd4b27f473a37">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># variance of estimated coef, beta hat</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>covbeta <span class="ot">&lt;-</span> tlc_gls<span class="sc">$</span>varBeta</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>beta <span class="ot">&lt;-</span> <span class="fu">coef</span>(tlc_gls)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># testing interactions together, manually</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>trt_week_coef <span class="ot">&lt;-</span> beta[<span class="dv">6</span><span class="sc">:</span><span class="dv">8</span>]</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>trt_week_cov <span class="ot">&lt;-</span> covbeta[<span class="dv">6</span><span class="sc">:</span><span class="dv">8</span>, <span class="dv">6</span><span class="sc">:</span><span class="dv">8</span>]</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># t(trt_week_coef) %*% solve(trt_week_cov) %*% trt_week_coef</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># avg treatment - avg control, H0: S = P</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>trt_L <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>) <span class="sc">-</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">matrix</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()<span class="co"># c(0,4,0,0,0,1,1,1)</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="co"># manually, its a scalar</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co"># trt_L[c(2, 6:8)] %*% beta[c(2, 6:8)] * </span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   solve(trt_L[,c(2, 6:8)] %*% covbeta[c(2, 6:8), c(2, 6:8)] %*% cbind(trt_L[,c(2, 6:8)])) * </span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   trt_L[c(2, 6:8)] %*% beta[c(2, 6:8)]</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="co"># average by week, H0: w0 = w1 = w4 = w6</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>w0_L <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>w1_L <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>w4_L <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>w6_L <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>week_L <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(w1_L <span class="sc">-</span> w0_L,</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>                   w4_L <span class="sc">-</span> w0_L,</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>                   w6_L <span class="sc">-</span> w0_L), <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">byrow=</span>T)</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the code above</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="fu">anova</span>(tlc_gls, <span class="at">L =</span> trt_L), <span class="co"># Wald F value = 25.43</span></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>      <span class="fu">anova</span>(tlc_gls, <span class="at">L =</span> week_L), <span class="co"># Wald F value = 61.49</span></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">anova</span>(tlc_gls, <span class="at">Terms =</span> <span class="dv">4</span>)) <span class="sc">%&gt;%</span>  <span class="co"># Wald F value = 35.9</span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Chisq =</span> numDF <span class="sc">*</span> <span class="st">`</span><span class="at">F-value</span><span class="st">`</span>,</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>                <span class="at">.after =</span> <span class="st">`</span><span class="at">F-value</span><span class="st">`</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">source =</span> <span class="fu">c</span>(<span class="st">"Trt"</span>, <span class="st">"Week"</span>, <span class="st">"Trt x Week"</span>), <span class="at">.before =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="r-anova.gls" class="level3" data-number="17.1.3">
<h3 data-number="17.1.3" class="anchored" data-anchor-id="r-anova.gls"><span class="header-section-number">17.1.3</span> R: anova.gls</h3>
<p>I’m not entirely sure what’s going on in the anova function in nlme. There is a value for intercept and treatment, and I’m not sure how to interpret that.</p>
<p>It seems the week and interaction are being calculated correctly, not sure why the treatment is different from the manual calculations above. In the presence of an interaction effect however, it doesn’t really matter.</p>
<p>Also notice that Chisq statistic is just the F statistic * ndf. Assuming chisq gives more liberal estimates, effectively infinite residual degrees of freedom.</p>
<p>The following are two anova tables of the same model,</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-8_e4976ea0905a4ba037e4fbdc89038e04">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># these results match</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(tlc_gls, <span class="at">type =</span> <span class="st">"marginal"</span>) <span class="co"># nlme</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Denom. DF: 392 
            numDF   F-value p-value
(Intercept)     1 1368.0793  &lt;.0001
trt             1    0.0712  0.7898
week            3    3.8731  0.0095
trt:week        3   35.9293  &lt;.0001</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Anova</span>(tlc_gls, <span class="st">"III"</span>, <span class="at">test.statistic =</span> <span class="st">"F"</span>, <span class="at">error.df =</span> tlc_gls<span class="sc">$</span>dims<span class="sc">$</span>N <span class="sc">-</span> tlc_gls<span class="sc">$</span>dims<span class="sc">$</span>p) <span class="co"># car w/ 392 error df</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Deviance Table (Type III tests)

Response: lead
             Df         F  Pr(&gt;F)    
(Intercept)   1 1368.0793 &lt; 2e-16 ***
trt           1    0.0712 0.78977    
week          3    3.8731 0.00946 ** 
trt:week      3   35.9293 &lt; 2e-16 ***
Residuals   392                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-9_b3a502e725308e1c9b92187a355d3810">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r fold-show code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># more liberal</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Anova</span>(tlc_gls, <span class="st">"III"</span>, <span class="at">test.statistic =</span> <span class="st">"Chisq"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Deviance Table (Type III tests)

Response: lead
            Df     Chisq Pr(&gt;Chisq)    
(Intercept)  1 1368.0793  &lt; 2.2e-16 ***
trt          1    0.0712   0.789625    
week         3   11.6192   0.008808 ** 
trt:week     3  107.7880  &lt; 2.2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-10_663123f614f96eb6eb2e0101f7a95be0">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># not sure what's going on here</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(tlc_gls, <span class="at">type =</span> <span class="st">"sequential"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="r-lm" class="level3" data-number="17.1.4">
<h3 data-number="17.1.4" class="anchored" data-anchor-id="r-lm"><span class="header-section-number">17.1.4</span> R: lm</h3>
<p>For comparison, we can see that gls gives the same estimates as the gls model, for profile curves, it basically goes through the mean of each group. lm is NOT the right way to do this analysis, but we want to see what it shows for comparison.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-11_18147f731a3be05b6c8a6004977403e8">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>tlc_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(lead <span class="sc">~</span> trt<span class="sc">*</span>week, <span class="at">data =</span> tlc)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="st">"gls"</span> <span class="ot">=</span> <span class="fu">coef</span>(tlc_gls), <span class="st">"lm"</span> <span class="ot">=</span> <span class="fu">coef</span>(tlc_lm))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                gls      lm
(Intercept)  26.272  26.272
trtA          0.268   0.268
weekw1       -1.612  -1.612
weekw4       -2.202  -2.202
weekw6       -2.626  -2.626
trtA:weekw1 -11.406 -11.406
trtA:weekw4  -8.824  -8.824
trtA:weekw6  -3.152  -3.152</code></pre>
</div>
</div>
<p>compare the standard errors of the estimates though</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-12_d48d3c287553099c27d07de23adc14b5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="st">"gls"</span> <span class="ot">=</span> <span class="fu">sqrt</span>(<span class="fu">diag</span>(<span class="fu">as.matrix</span>(tlc_gls<span class="sc">$</span>varBeta))),</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>      <span class="st">"lm"</span> <span class="ot">=</span> tlc_lm <span class="sc">%&gt;%</span> <span class="fu">tidy</span>() <span class="sc">%&gt;%</span> <span class="fu">pull</span>(std.error))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                  gls        lm
(Intercept) 0.7102929 0.9370175
trtA        1.0045059 1.3251428
weekw1      0.7919199 1.3251428
weekw4      0.8149021 1.3251428
weekw6      0.8885253 1.3251428
trtA:weekw1 1.1199438 1.8740349
trtA:weekw4 1.1524456 1.8740349
trtA:weekw6 1.2565645 1.8740349</code></pre>
</div>
</div>
<p>not quite sure this is the fairest comparison, but the heterogeneity in the estimated variance seems to have stabilized the standardized residuals. Need to double check if this is the right stabilization.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-13_b6a752c403b8a24ab6d50f7f96e8ea18">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tlc_gls, <span class="fu">resid</span>(., <span class="at">type =</span> <span class="st">"pearson"</span>) <span class="sc">~</span> week_int) <span class="co"># show against time (instead of default fitted)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tlc<span class="sc">$</span>week_int, <span class="fu">rstudent</span>(tlc_lm))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="r-auc" class="level3" data-number="17.1.5">
<h3 data-number="17.1.5" class="anchored" data-anchor-id="r-auc"><span class="header-section-number">17.1.5</span> R: AUC</h3>
<p>A third strategy is to do summary statistics of the curves. The section in the book refers to creating 1 df tests, which are more powerful than the overall F test of trt x time interaction. Often times, if you measure many many time points, the overall f test becomes diluted and it becomes more difficult to detect differences in the two curves, so these are designed to have more power than those tests. <em>should only be specified prior to analysis though, to keep with proper significance control</em></p>
<p>We consider two tests,</p>
<ol type="1">
<li>average difference minus baseline.</li>
</ol>
<ul>
<li>the contrast will take the form (where <span class="math inline">n</span> is the number of occassions)</li>
</ul>
<p><span class="math display">
\begin{aligned}
L &amp;= (L_1, -L_1) \\
L_1 &amp;= \left(-1, \frac{1}{n-1}, \frac{1}{n-1}, \dots, \frac{1}{n-1}\right)
\end{aligned}
</span></p>
<ol start="2" type="1">
<li>Area under curve (AUC) minus baseline</li>
</ol>
<ul>
<li>approximated by trapezoids</li>
</ul>
<p>both of these can be formulated as a contrast, we’ll use emmeans for convenience here.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-15_6088a49c365a70c06ea9919c44721bd6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>emm_tlc_gls <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(tlc_gls, <span class="at">specs=</span><span class="fu">c</span>(<span class="st">"week"</span>, <span class="st">"trt"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Analytical Satterthwaite method not available; using appx-satterthwaite</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>emm_tlc_gls</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> week trt emmean    SE   df lower.CL upper.CL
 w0   P     26.3 0.710 97.9     24.9     27.7
 w1   P     24.7 0.942 98.1     22.8     26.5
 w4   P     24.1 0.973 98.1     22.1     26.0
 w6   P     23.6 1.083 98.1     21.5     25.8
 w0   A     26.5 0.710 97.9     25.1     27.9
 w1   A     13.5 0.942 98.1     11.7     15.4
 w4   A     15.5 0.973 98.1     13.6     17.4
 w6   A     20.8 1.083 98.1     18.6     22.9

Degrees-of-freedom method: appx-satterthwaite 
Confidence level used: 0.95 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>avg_minus_baseline_L <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>, <span class="dv">1</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>, <span class="sc">-</span><span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>auc_minus_baseline_L <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">5.5</span>, <span class="sc">-</span><span class="dv">2</span>, <span class="sc">-</span><span class="fl">2.5</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="sc">-</span><span class="fl">5.5</span>, <span class="dv">2</span>, <span class="fl">2.5</span>, <span class="dv">1</span>)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">contrast</span>(emm_tlc_gls, <span class="fu">list</span>(<span class="st">"avg minus baseline"</span> <span class="ot">=</span> avg_minus_baseline_L,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"AUC minus baseline"</span> <span class="ot">=</span> auc_minus_baseline_L))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> contrast           estimate   SE   df t.ratio p.value
 avg minus baseline     7.79 0.95 98.1   8.205  &lt;.0001
 AUC minus baseline   -48.02 5.35 97.9  -8.973  &lt;.0001

Degrees-of-freedom method: appx-satterthwaite </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(tlc_gls, <span class="at">L =</span> avg_minus_baseline_L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Denom. DF: 392 
 F-test for linear combination(s)
(Intercept)        trtA      weekw1      weekw4      weekw6 trtA:weekw1 
 -1.0000000   0.3333333   0.3333333   0.3333333   1.0000000  -0.3333333 
trtA:weekw4 trtA:weekw6 
 -0.3333333  -0.3333333 
  numDF F-value p-value
1     1 91.2127  &lt;.0001</code></pre>
</div>
</div>
</section>
</section>
<section id="example-ratdrink" class="level2 tabset" data-number="17.2">
<h2 class="tabset anchored" data-number="17.2" data-anchor-id="example-ratdrink"><span class="header-section-number">17.2</span> Example: Ratdrink</h2>
<p>The following example is from <a href="#References">@faraway_extending_2016</a></p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-16_86020ba37d14bc124ab9bb83bb919356">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ratdrink)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>ratdrink <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(weeks, wt, <span class="at">group =</span> subject)) <span class="sc">+</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>treat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="different-intercepts" class="level3" data-number="17.2.1">
<h3 data-number="17.2.1" class="anchored" data-anchor-id="different-intercepts"><span class="header-section-number">17.2.1</span> different intercepts</h3>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-17_16300df22155985f764738c08f2bfbbe">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># full lm</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>rat_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(wt <span class="sc">~</span> weeks<span class="sc">*</span>treat <span class="sc">+</span> subject, <span class="at">data =</span> ratdrink) <span class="co"># fixed "block"</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>rat_mmer <span class="ot">&lt;-</span> <span class="fu">lmer</span>(wt <span class="sc">~</span> weeks<span class="sc">*</span>treat <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>subject), <span class="at">data =</span> ratdrink) <span class="co"># random "block"</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="co"># plotting the predictions</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>rat_preds <span class="ot">&lt;-</span> ratdrink <span class="sc">%&gt;%</span> <span class="fu">add_column</span>(<span class="at">lm_yhat =</span> <span class="fu">predict</span>(rat_lm),</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">mmer_yhat =</span> <span class="fu">predict</span>(rat_mmer))</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>rat_preds <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="fu">c</span>(wt, lm_yhat, mmer_yhat), <span class="at">names_to =</span> <span class="st">"response"</span>, <span class="at">values_to =</span> <span class="st">"y"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(weeks, y, <span class="at">groups =</span> subject)) <span class="sc">+</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(response<span class="sc">~</span>treat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="different-slopes" class="level3" data-number="17.2.2">
<h3 data-number="17.2.2" class="anchored" data-anchor-id="different-slopes"><span class="header-section-number">17.2.2</span> different slopes</h3>
<p>The bottom row is the raw data while, the second row is the predictions from a random subject, and finally the top row is the predictions from the fully fixed model.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-18_822445a527ce1f2dd049c07a044d6eed">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>rat_lm_diffslope <span class="ot">&lt;-</span> <span class="fu">lm</span>(wt<span class="sc">~</span>weeks<span class="sc">*</span>treat <span class="sc">+</span> subject <span class="sc">+</span> subject<span class="sc">:</span>weeks, <span class="at">data =</span> ratdrink)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="co"># rat_mmer_interaction &lt;- lmer(wt ~ weeks*treat + (1|weeks:subject), data = ratdrink) # weeks is continuous! doesn't work.... too many random effect intercepts</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>rat_mmer_interaction <span class="ot">&lt;-</span> <span class="fu">lmer</span>(wt <span class="sc">~</span> weeks<span class="sc">*</span>treat <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>subject) <span class="sc">+</span> subject<span class="sc">:</span>weeks, <span class="at">data =</span> ratdrink) <span class="co"># weird model....b/c subject:weeks is fixed not sure when </span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>rat_mmer_random_slope <span class="ot">&lt;-</span> <span class="fu">lmer</span>(wt <span class="sc">~</span> weeks<span class="sc">*</span>treat <span class="sc">+</span> (weeks<span class="sc">|</span>subject), <span class="at">data =</span> ratdrink)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>rat_mmer_random_slope_nocor <span class="ot">&lt;-</span> <span class="fu">lmer</span>(wt <span class="sc">~</span> weeks<span class="sc">*</span>treat <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> subject) <span class="sc">+</span> (<span class="dv">0</span> <span class="sc">+</span> weeks <span class="sc">||</span> subject), <span class="at">data =</span> ratdrink) <span class="co"># without correlation between intercept and slope</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>rat_preds_slope <span class="ot">&lt;-</span> ratdrink <span class="sc">%&gt;%</span> <span class="fu">add_column</span>(<span class="at">lm_diffslope_yhat =</span> <span class="fu">predict</span>(rat_lm_diffslope),</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">mmer_interaction_yhat =</span> <span class="fu">predict</span>(rat_mmer_interaction),</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">mmer_random_slope_yhat =</span> <span class="fu">predict</span>(rat_mmer_random_slope),</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">mmer_random_slope_nocor_yhat =</span> <span class="fu">predict</span>(rat_mmer_random_slope_nocor))</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>rat_preds_slope <span class="sc">%&gt;%</span> </span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">c</span>(wt, </span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>                 lm_diffslope_yhat, </span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>                 mmer_interaction_yhat, </span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>                 mmer_random_slope_yhat, </span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>                 mmer_random_slope_nocor_yhat), <span class="at">names_to =</span> <span class="st">"response"</span>, <span class="at">values_to =</span> <span class="st">"y"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(weeks, y, <span class="at">group =</span> subject)) <span class="sc">+</span></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(response<span class="sc">~</span>treat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-19_9a512d095fb83b7b1090cda9193fc2ca">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># profile conf intervals</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">confint</span>(rat_mmer_random_slope)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Computing profile confidence intervals ...</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>                            2.5 %     97.5 %
.sig01                  3.4506344  7.6654748
.sig02                 -0.5261030  0.3794203
.sig03                  2.6064121  4.8687653
.sigma                  3.7555591  5.1142342
(Intercept)            48.8692859 56.8907140
weeks                  24.0547424 28.9052555
treatthiouracil        -0.8920062 10.4520061
treatthyroxine         -7.0445321  5.4559606
weeks:treatthiouracil -12.7998322 -5.9401707
weeks:treatthyroxine   -3.1166339  4.4423449</code></pre>
</div>
</div>
</section>
</section>
<section id="example-body-fat-and-menarche" class="level2" data-number="17.3">
<h2 data-number="17.3" class="anchored" data-anchor-id="example-body-fat-and-menarche"><span class="header-section-number">17.3</span> Example: Body Fat and Menarche</h2>
<p>We use the MIT growth study menarche example for this section, from <a href="https://content.sph.harvard.edu/fitzmaur/ala2e/">Fitzmaurice website</a></p>
<p>Covariates/Response:</p>
<ul>
<li>id : girl id</li>
<li>age : age at observation</li>
<li>agemen : age of menarche for girl</li>
<li>time : age - agemen, time relative to menarche.</li>
<li>pbf : percentage body fat</li>
</ul>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-20_5b8ad96f281c8640088909fc16809797">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>fat <span class="ot">&lt;-</span> <span class="fu">read.dta</span>(<span class="st">"data/fat.dta"</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co"># select 20 random girls and show response curve</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>fat <span class="sc">%&gt;%</span> </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_nest</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n=</span><span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(data) <span class="sc">%&gt;%</span> </span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(time, pbf)) <span class="sc">+</span> </span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>id) <span class="sc">+</span></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="modeling" class="level3" data-number="17.3.1">
<h3 data-number="17.3.1" class="anchored" data-anchor-id="modeling"><span class="header-section-number">17.3.1</span> Modeling</h3>
<p><span class="citation" data-cites="fitzmaurice_applied_2011">Fitzmaurice, Laird, and Ware (<a href="#ref-fitzmaurice_applied_2011" role="doc-biblioref">2011</a>)</span> in chapter 8.8 analyzes this as a piecewise random effects model.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-21_f762c93ef2b534474094181648eb9dc1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># to fit piecewise function, need variable with after menarche time</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>fat_post <span class="ot">&lt;-</span> fat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">timepost =</span> time <span class="sc">*</span> (time <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="co"># create variable for post menarche time</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>fat_lme <span class="ot">&lt;-</span> <span class="fu">lme</span>(pbf <span class="sc">~</span> time <span class="sc">+</span> timepost,</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">random =</span> <span class="sc">~</span>time <span class="sc">+</span> timepost <span class="sc">|</span> id,</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> fat_post)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The fixed effects of the model:</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-22_dcbb64f4317bda8f3acbf6a26b863897">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fixed effects</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>fat_lme <span class="sc">%&gt;%</span> <span class="fu">tidy</span>() <span class="sc">%&gt;%</span> <span class="fu">filter</span>(effect <span class="sc">==</span> <span class="st">"fixed"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(term, estimate, std.error, statistic, df, p.value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 6
  term        estimate std.error statistic    df   p.value
  &lt;chr&gt;          &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;
1 (Intercept)   21.4       0.565     37.8    885 3.95e-187
2 time           0.417     0.157      2.65   885 8.09e-  3
3 timepost       2.05      0.228      8.98   885 1.60e- 18</code></pre>
</div>
</div>
<p>The random effects and standard errors calculated from inverse expected fisher information (with package <code>lmeInfo</code>)</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-23_b1a86c1c971525427f52956f519a7d59">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stats.oarc.ucla.edu/r/faq/how-can-i-calculate-standard-errors-for-variance-components-from-mixed-models/</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Random effects and standard errors</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>fat_varcomp <span class="ot">&lt;-</span> <span class="fu">getVarCov</span>(fat_lme)[<span class="fu">upper.tri</span>(<span class="fu">getVarCov</span>(fat_lme), <span class="at">diag =</span> <span class="cn">TRUE</span>)]</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="at">estimate =</span> fat_varcomp,</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>      <span class="st">`</span><span class="at">std. err</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sqrt</span>(<span class="fu">diag</span>(<span class="fu">varcomp_vcov</span>(fat_lme)))[<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                                  estimate  std. err
Tau.id.var((Intercept))          45.939153 5.7556122
Tau.id.cov(time,(Intercept))      2.526047 1.1984610
Tau.id.var(time)                  1.630971 0.4023225
Tau.id.cov(timepost,(Intercept)) -6.109078 1.8486864
Tau.id.cov(timepost,time)        -1.750363 0.5746242
Tau.id.var(timepost)              2.749384 0.9273656</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="testing-for-heteroskedasticiy-and-autocorrelation" class="level1" data-number="18">
<h1 data-number="18"><span class="header-section-number">18</span> Testing for heteroskedasticiy and autocorrelation</h1>
<ul>
<li>Goldfeld-Quandt - 2 subsamples with possibly different variances.</li>
<li>Breusch-Pagan Test - for conditional heteroskedasticity</li>
<li>White Test - more generalized test, including exogenous variables but also polynomial and interaction terms</li>
<li>Breush-Godfrey Test -</li>
<li>Durbin Watson Test</li>
</ul>
<p>There’s - <code>strucchange::gefp</code> - structural change testing.</p>
<section id="resources" class="level2" data-number="18.1">
<h2 data-number="18.1" class="anchored" data-anchor-id="resources"><span class="header-section-number">18.1</span> Resources</h2>
<ul>
<li><a href="http://web.vu.lt/mif/a.buteikis/wp-content/uploads/PE_Book/4-7-Multiple-heteroskedastic.html">Practical Econometrics Book</a></li>
</ul>
</section>
</section>
<section id="modeling-the-covariance" class="level1" data-number="19">
<h1 data-number="19"><span class="header-section-number">19</span> Modeling the Covariance</h1>
<p>We should note that there are econometric approaches, such as sandwich estimators, and mixed model approaches that are all competing in the space of modeling the variance. Broadly there are three approaches for variance modeling:</p>
<ol type="1">
<li>random effects (“lme4”)</li>
<li>generalized estimating equations (“gee”)</li>
<li>feasible generalized least squares (“plm”)</li>
</ol>
<p>Packages with this approach from the <a href="https://cran.r-project.org/web/packages/sandwich/vignettes/sandwich-CL.pdf">sandwich vignette</a></p>
<ul>
<li>“sandwich”</li>
<li>“multiwayvcov” - mostly lm or glm like objects</li>
<li>“plm” - several sandwich covariances for panel and fixed effect linear models.</li>
<li>“geepack” - for glm type models, and subsequent geeglm objects</li>
<li>“clubSandwich” - orginary or weighted least sqaures regression</li>
<li>“clusterSEs” - ordinary or weighted least squares regression</li>
<li>“lfe” - standard models</li>
</ul>
<section id="examples" class="level2 tabset" data-number="19.1">
<h2 class="tabset anchored" data-number="19.1" data-anchor-id="examples"><span class="header-section-number">19.1</span> Examples</h2>
<section id="example-spruce-trees" class="level3" data-number="19.1.1">
<h3 data-number="19.1.1" class="anchored" data-anchor-id="example-spruce-trees"><span class="header-section-number">19.1.1</span> Example: Spruce Trees</h3>
<ul>
<li>y: response (log of product of tree size and diameter squared)</li>
<li>tx: treatment ( 2 levels )
<ul>
<li>0: control condition (25 trees per time point)</li>
<li>1: ozone exposure at 70 ppb (54 trees per time point)</li>
</ul></li>
<li>day: time (days since 1988 Jan 1)</li>
<li>chamber: block (ozone controlled chamber)</li>
</ul>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-24_7f5ce801bdbaec4c04f89c37f414708a">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># str(spruce88)</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="co"># xtabs(~day + tx, data=spruce88)</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="co"># xtabs(~tx,data = spruce88)</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Example of selecting out some groups of elements for longitudinal analysis. Used above for highlighting</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>spruce88 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(id <span class="sc">%in%</span> <span class="fu">sample</span>(<span class="fu">levels</span>(spruce88[,<span class="st">"id"</span>])))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] y       day     tx      chamber id     
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>sid <span class="ot">&lt;-</span>spruce88 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(chamber) <span class="sc">%&gt;%</span> <span class="fu">sample_n</span>(<span class="dv">5</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>spruce_highlight <span class="ot">&lt;-</span> spruce88 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(id <span class="sc">%in%</span> sid[[<span class="st">"id"</span>]])</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Emphasizing lines in the plot with background lighter</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>g_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> spruce88, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>day, <span class="at">y=</span>y, <span class="at">color =</span> <span class="fu">factor</span>(tx), <span class="at">group=</span><span class="fu">factor</span>(id))) <span class="sc">+</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>chamber)</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>(g_base <span class="sc">+</span></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>   <span class="fu">geom_point</span>(<span class="at">alpha=</span>.<span class="dv">2</span>)<span class="sc">+</span> <span class="fu">geom_line</span>(<span class="at">alpha=</span>.<span class="dv">2</span>) <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="at">data=</span>spruce_highlight, <span class="fu">aes</span>(<span class="at">x=</span>day, <span class="at">y=</span>y, <span class="at">group=</span><span class="fu">factor</span>(id))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># (g_base + geom_point(alpha=.2) + geom_smooth(span=.5, aes(group="abc")))</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: using a constant for the group, will override the previous grouping.</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Also, the smoothing will come up with "singularities" since there are only 5 distinct x data points, and lowess will have trouble.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="exploring-correlation-from-week-to-week-and-getting-correlation-structure." class="level3" data-number="19.1.2">
<h3 data-number="19.1.2" class="anchored" data-anchor-id="exploring-correlation-from-week-to-week-and-getting-correlation-structure."><span class="header-section-number">19.1.2</span> Exploring correlation from week to week, and getting correlation structure.</h3>
<p>Suggestion if the data is taken at many different points, just round it to the nearest year and we can still get some sense of the correlation over time. In the case everything is discrete, we can just use the data categories as is.</p>
<p>We should remove the effect of the means from each week. The suggestion in the book is to remove the covariate effect by fitting a regression on the data.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-25_fffeb54e1d663875d47bdb476cb95d23">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove the effect of means from each week</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># residuals(lm(y~day + tx + chamber, data=spruce88))</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(spruce88)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   395 obs. of  5 variables:
 $ y      : num  4.51 4.98 5.41 5.9 6.14 ...
 $ day    : num  -49 -27 0 26 57 -49 -27 0 26 57 ...
 $ tx     : num  1 1 1 1 1 1 1 1 1 1 ...
 $ chamber: num  1 1 1 1 1 1 1 1 1 1 ...
 $ id     : int  1 1 1 1 1 2 2 2 2 2 ...</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>spruce_resid <span class="ot">&lt;-</span> spruce88 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">resid =</span> <span class="fu">residuals</span>(<span class="fu">lm</span>(y<span class="sc">~</span>day <span class="sc">+</span> tx <span class="sc">+</span> chamber)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="example-tlc-1" class="level3" data-number="19.1.3">
<h3 data-number="19.1.3" class="anchored" data-anchor-id="example-tlc-1"><span class="header-section-number">19.1.3</span> Example: TLC</h3>
<p>For exploratory analysis, we should show the unstructured estimated covariance matrix from the model, shown for an individual <span class="math inline">i</span>,</p>
<p><span class="math display">
\begin{aligned}
Y_i = X_i\beta + \varepsilon_i
\end{aligned}
</span></p>
<p>assuming that <span class="math inline">\varepsilon_i \sim N(0,\Sigma)</span>. We show the estimate <span class="math inline">\hat \Sigma</span> as unstructured covariance matrix:</p>
<p>This is table 5.3 in <span class="citation" data-cites="fitzmaurice_applied_2011">Fitzmaurice, Laird, and Ware (<a href="#ref-fitzmaurice_applied_2011" role="doc-biblioref">2011</a>)</span>.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-26_1b54d9079c474ae861585d9c8ac2a870">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># direct group means covariance</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>tlc_gls_un_cov <span class="ot">&lt;-</span> <span class="fu">gls</span>(lead <span class="sc">~</span> trt<span class="sc">*</span>week,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">correlation=</span><span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> time <span class="sc">|</span> id), </span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> week), </span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data=</span>tlc)</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(tlc_gls_un_cov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Marginal variance covariance matrix
       [,1]   [,2]   [,3]   [,4]
[1,] 25.226 19.107 19.700 22.202
[2,] 19.107 44.346 35.535 29.675
[3,] 19.700 35.535 47.377 30.620
[4,] 22.202 29.675 30.620 58.651
  Standard Deviations: 5.0225 6.6593 6.8831 7.6584 </code></pre>
</div>
</div>
<p>We can also look at the correlation with a scatter plot.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-27_a1c2e28ca7e3850658037650babcd91d">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>tlc[<span class="st">"week"</span>] <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(tlc<span class="sc">$</span>week)</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Split into the exposed and placebo groups and make plots from the placebo group</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>tlc_exposed <span class="ot">&lt;-</span> tlc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">"A"</span>) <span class="sc">%&gt;%</span> <span class="fu">spread</span>(week, lead)</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>tlc_placebo <span class="ot">&lt;-</span> tlc <span class="sc">%&gt;%</span> <span class="fu">filter</span>(trt <span class="sc">==</span> <span class="st">"P"</span>) <span class="sc">%&gt;%</span> <span class="fu">spread</span>(week, lead)</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(tlc_placebo[<span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(tlc_placebo[<span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>          week_int      time w0 w1
week_int 1.0000000 0.9844952 NA NA
time     0.9844952 1.0000000 NA NA
w0              NA        NA  1 NA
w1              NA        NA NA  1</code></pre>
</div>
</div>
</section>
<section id="example-body-fat-and-menarche-1" class="level3" data-number="19.1.4">
<h3 data-number="19.1.4" class="anchored" data-anchor-id="example-body-fat-and-menarche-1"><span class="header-section-number">19.1.4</span> Example: Body Fat and Menarche</h3>
<p>The variance covariance matrix can be extracted in two ways:</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-28_4afe036f0af592fb855dcae7f258026a">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># using reStruct from model is scale version of covariance matrix</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">list</span>(<span class="fu">as.matrix</span>(fat_lme<span class="sc">$</span>modelStruct<span class="sc">$</span>reStruct[[<span class="dv">1</span>]]) <span class="sc">*</span> fat_lme<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span>,</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>     <span class="fu">getVarCov</span>(fat_lme))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]
            (Intercept)      time  timepost
(Intercept)   45.939153  2.526047 -6.109078
time           2.526047  1.630971 -1.750363
timepost      -6.109078 -1.750363  2.749384

[[2]]
Random effects variance covariance matrix
            (Intercept)    time timepost
(Intercept)     45.9390  2.5260  -6.1091
time             2.5260  1.6310  -1.7504
timepost        -6.1091 -1.7504   2.7494
  Standard Deviations: 6.7778 1.2771 1.6581 </code></pre>
</div>
</div>
<p>We can also model the within correlation differently with <code>correlation</code></p>
<p><code>corCAR1</code> - models the correlation by <code>phi = .2</code> (default), has autocorrelation function <span class="math inline">h(\cdot)</span>, with parameters <span class="math inline">s</span> distance, and <span class="math inline">\phi</span> correlation</p>
<p><span class="math display">
\begin{aligned}
h(s, \phi) = \phi^s \quad s \geq 0, \phi \geq 0
\end{aligned}
</span></p>
<p>We can manually create the correlation matrix using this function and pairwise distances, or use the corStruct class in <code>nlme</code>.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-29_9b14e40a765250703752038805e4ecf2">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">list</span>(<span class="at">manual =</span> .<span class="dv">2</span><span class="sc">^</span><span class="fu">dist</span>(fat_post<span class="sc">$</span>time[fat_post<span class="sc">$</span>id <span class="sc">==</span> <span class="dv">1</span>]),</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">lme =</span> <span class="fu">corMatrix</span>(<span class="fu">Initialize</span>(<span class="fu">corCAR1</span>(<span class="at">form =</span> <span class="sc">~</span>time <span class="sc">|</span> id), <span class="at">data =</span> fat_post))[[<span class="dv">1</span>]]) <span class="co"># phi = .2 default</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$manual
             1            2            3            4            5
2 0.1968068917                                                    
3 0.0454964703 0.2311731563                                       
4 0.0098617995 0.0501090149 0.2167596607                          
5 0.0018198587 0.0092469255 0.0399999969 0.1845361667             
6 0.0003639718 0.0018493853 0.0080000000 0.0369072362 0.2000000156

$lme
             [,1]        [,2]       [,3]        [,4]        [,5]         [,6]
[1,] 1.0000000000 0.196806892 0.04549647 0.009861799 0.001819859 0.0003639718
[2,] 0.1968068917 1.000000000 0.23117316 0.050109015 0.009246926 0.0018493853
[3,] 0.0454964703 0.231173156 1.00000000 0.216759661 0.039999997 0.0080000000
[4,] 0.0098617995 0.050109015 0.21675966 1.000000000 0.184536167 0.0369072362
[5,] 0.0018198587 0.009246926 0.04000000 0.184536167 1.000000000 0.2000000156
[6,] 0.0003639718 0.001849385 0.00800000 0.036907236 0.200000016 1.0000000000</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-30_35eefae2b4a4943c00f3bcc6511974e7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>fat_car1 <span class="ot">&lt;-</span> <span class="fu">lme</span>(pbf<span class="sc">~</span>time <span class="sc">+</span> timepost,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> id,</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">corr=</span><span class="fu">corCAR1</span>(,<span class="at">form=</span> <span class="sc">~</span> time <span class="sc">|</span> id),</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> fat_post)</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>fat_car1 <span class="sc">%&gt;%</span> <span class="fu">getVarCov</span>(<span class="at">type =</span> <span class="st">"marginal"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>id 1 
Marginal variance covariance matrix
       1      2      3      4      5      6
1 48.672 39.818 35.798 33.626 32.441 31.897
2 39.818 48.672 40.439 35.991 33.563 32.448
3 35.798 40.439 48.672 40.185 35.554 33.427
4 33.626 35.991 40.185 48.672 39.581 35.408
5 32.441 33.563 35.554 39.581 48.672 39.878
6 31.897 32.448 33.427 35.408 39.878 48.672
  Standard Deviations: 6.9765 6.9765 6.9765 6.9765 6.9765 6.9765 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>fat_lme <span class="sc">%&gt;%</span> <span class="fu">getVarCov</span>(<span class="at">type =</span> <span class="st">"marginal"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>id 1 
Marginal variance covariance matrix
       1      2      3      4      5      6
1 60.288 46.991 43.546 39.949 36.007 32.886
2 46.991 54.304 42.885 40.853 38.553 35.311
3 43.546 42.885 51.763 41.668 40.846 37.496
4 39.949 40.853 41.668 51.991 43.240 39.776
5 36.007 38.553 40.846 43.240 55.056 42.044
6 32.886 35.311 37.496 39.776 42.044 48.858
  Standard Deviations: 7.7645 7.3691 7.1946 7.2105 7.42 6.9898 </code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-31_0a35fc0550c50cf68677e3a924220c0a">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(fat_car1, <span class="at">type =</span> <span class="st">"marginal"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>id 1 
Marginal variance covariance matrix
       1      2      3      4      5      6
1 48.672 39.818 35.798 33.626 32.441 31.897
2 39.818 48.672 40.439 35.991 33.563 32.448
3 35.798 40.439 48.672 40.185 35.554 33.427
4 33.626 35.991 40.185 48.672 39.581 35.408
5 32.441 33.563 35.554 39.581 48.672 39.878
6 31.897 32.448 33.427 35.408 39.878 48.672
  Standard Deviations: 6.9765 6.9765 6.9765 6.9765 6.9765 6.9765 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(fat_car1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>id = pdLogChol(1) 
            Variance StdDev  
(Intercept) 31.37028 5.600917
Residual    17.30157 4.159516</code></pre>
</div>
</div>
</section>
<section id="example-dental" class="level3" data-number="19.1.5">
<h3 data-number="19.1.5" class="anchored" data-anchor-id="example-dental"><span class="header-section-number">19.1.5</span> Example: Dental</h3>
<p>This section mostly uses <code>gls</code> with a variety of different covariance matrices.</p>
<p>This dataset shows measuruments of pituitary gland to pteryomaxillary fissure. It has 11 girls and 16 boys at ages 8, 10, 12, 14.</p>
<ul>
<li>id : patient id</li>
<li>gender: male/female</li>
<li>distance: the response variable</li>
</ul>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-32_7c72fa75651f327f08add9f795752b9e">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>dental_wide <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/dental.txt"</span>,</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">header =</span> <span class="cn">FALSE</span>,</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">"id"</span>, <span class="st">"gender"</span>, <span class="st">"y1"</span>, <span class="st">"y2"</span>, <span class="st">"y3"</span>, <span class="st">"y4"</span>))</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="co"># long data format</span></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>dental <span class="ot">&lt;-</span> dental_wide <span class="sc">%&gt;%</span> </span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> y1<span class="sc">:</span>y4, </span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"age"</span>,</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"distance"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">id =</span> <span class="fu">factor</span>(id),</span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">age =</span> <span class="fu">recode</span>(age,</span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y1 =</span> <span class="dv">8</span>,</span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y2 =</span> <span class="dv">10</span>,</span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y3 =</span> <span class="dv">12</span>,</span>
<span id="cb68-16"><a href="#cb68-16" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y4 =</span> <span class="dv">14</span>),</span>
<span id="cb68-17"><a href="#cb68-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">agef =</span> <span class="fu">factor</span>(age),</span>
<span id="cb68-18"><a href="#cb68-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">.after =</span> <span class="st">"age"</span>)</span>
<span id="cb68-19"><a href="#cb68-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-20"><a href="#cb68-20" aria-hidden="true" tabindex="-1"></a>dental_mean <span class="ot">&lt;-</span> dental <span class="sc">%&gt;%</span></span>
<span id="cb68-21"><a href="#cb68-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(age, gender) <span class="sc">%&gt;%</span> </span>
<span id="cb68-22"><a href="#cb68-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">avg_distance =</span> <span class="fu">mean</span>(distance),</span>
<span id="cb68-23"><a href="#cb68-23" aria-hidden="true" tabindex="-1"></a>                   <span class="at">.groups =</span> <span class="st">"drop_last"</span>)</span>
<span id="cb68-24"><a href="#cb68-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-25"><a href="#cb68-25" aria-hidden="true" tabindex="-1"></a>dental <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(age, distance, <span class="at">group =</span> id)) <span class="sc">+</span></span>
<span id="cb68-26"><a href="#cb68-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> .<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb68-27"><a href="#cb68-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">alpha =</span> .<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb68-28"><a href="#cb68-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data =</span> dental_mean, <span class="at">mapping =</span> <span class="fu">aes</span>(age, avg_distance, <span class="at">group =</span> <span class="cn">NULL</span>), <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span> </span>
<span id="cb68-29"><a href="#cb68-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gender)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-33_50886dd0a6c5bfd9f85d75a6cf83d24e">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># all fixed with id</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>dental_fixed <span class="ot">&lt;-</span> <span class="fu">lm</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender <span class="sc">+</span> id, <span class="at">data =</span> dental)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>dental_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender, <span class="at">data =</span> dental)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-34_e48b5f6547161ba80078e7fece357648">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># modeling covariance</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>dental_ident <span class="ot">&lt;-</span> <span class="fu">gls</span>(distance<span class="sc">~</span>age<span class="sc">*</span>gender, <span class="at">data =</span> dental) <span class="co"># same as lm</span></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="do">## heterogenous </span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>dental_het <span class="ot">&lt;-</span> <span class="fu">gls</span>(distance<span class="sc">~</span>age<span class="sc">*</span>gender,</span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental,</span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> age))</span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a><span class="do">## unstructured</span></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>dental_un <span class="ot">&lt;-</span> <span class="fu">gls</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender, </span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental,</span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> id))</span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a><span class="do">## heterogenous unstructured</span></span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>dental_hun <span class="ot">&lt;-</span> <span class="fu">gls</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender, </span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental,</span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> id),</span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> age))</span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a><span class="do">## compound symmetry</span></span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a>dental_cs <span class="ot">&lt;-</span> <span class="fu">gls</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender, </span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental,</span>
<span id="cb70-23"><a href="#cb70-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">correlation =</span> <span class="fu">corCompSymm</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> id))</span>
<span id="cb70-24"><a href="#cb70-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-25"><a href="#cb70-25" aria-hidden="true" tabindex="-1"></a><span class="do">## Heterogenous CS</span></span>
<span id="cb70-26"><a href="#cb70-26" aria-hidden="true" tabindex="-1"></a>dental_csh <span class="ot">&lt;-</span> <span class="fu">gls</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender,</span>
<span id="cb70-27"><a href="#cb70-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental,</span>
<span id="cb70-28"><a href="#cb70-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">correlation =</span> <span class="fu">corCompSymm</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> id),</span>
<span id="cb70-29"><a href="#cb70-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> age))</span>
<span id="cb70-30"><a href="#cb70-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-31"><a href="#cb70-31" aria-hidden="true" tabindex="-1"></a><span class="do">## Autoregressive</span></span>
<span id="cb70-32"><a href="#cb70-32" aria-hidden="true" tabindex="-1"></a>dental_ar <span class="ot">&lt;-</span> <span class="fu">gls</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender,</span>
<span id="cb70-33"><a href="#cb70-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental,</span>
<span id="cb70-34"><a href="#cb70-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">correlation =</span> <span class="fu">corAR1</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> id),</span>
<span id="cb70-35"><a href="#cb70-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> age))</span>
<span id="cb70-36"><a href="#cb70-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-37"><a href="#cb70-37" aria-hidden="true" tabindex="-1"></a><span class="do">## Autoregressive Heterogeneous</span></span>
<span id="cb70-38"><a href="#cb70-38" aria-hidden="true" tabindex="-1"></a>dental_arh <span class="ot">&lt;-</span> <span class="fu">gls</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender,</span>
<span id="cb70-39"><a href="#cb70-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental,</span>
<span id="cb70-40"><a href="#cb70-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">correlation =</span> <span class="fu">corAR1</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> id),</span>
<span id="cb70-41"><a href="#cb70-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> age))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-35_f06bca76bd9884c2dbe6b3bace4077d1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_un)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_hun)</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_cs)</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_csh)</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_ar)</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_arh)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Marginal variance covariance matrix
       [,1]   [,2]   [,3]   [,4]
[1,] 5.2300 3.0094 3.3345 2.6923
[2,] 3.0094 5.2300 3.0035 3.9166
[3,] 3.3345 3.0035 5.2300 3.7687
[4,] 2.6923 3.9166 3.7687 5.2300
  Standard Deviations: 2.2869 2.2869 2.2869 2.2869 
Marginal variance covariance matrix
       [,1]   [,2]   [,3]   [,4]
[1,] 5.4252 2.7092 3.8411 2.7152
[2,] 2.7092 4.1906 2.9745 3.3137
[3,] 3.8411 2.9745 6.2632 4.1333
[4,] 2.7152 3.3137 4.1333 4.9862
  Standard Deviations: 2.3292 2.0471 2.5026 2.233 
Marginal variance covariance matrix
       [,1]   [,2]   [,3]   [,4]
[1,] 5.2207 3.2986 3.2986 3.2986
[2,] 3.2986 5.2207 3.2986 3.2986
[3,] 3.2986 3.2986 5.2207 3.2986
[4,] 3.2986 3.2986 3.2986 5.2207
  Standard Deviations: 2.2849 2.2849 2.2849 2.2849 
Marginal variance covariance matrix
       [,1]   [,2]   [,3]   [,4]
[1,] 5.6967 3.1209 3.7419 3.3309
[2,] 3.1209 4.2365 3.2269 2.8724
[3,] 3.7419 3.2269 6.0901 3.4440
[4,] 3.3309 2.8724 3.4440 4.8256
  Standard Deviations: 2.3868 2.0583 2.4678 2.1967 
Marginal variance covariance matrix
       [,1]   [,2]   [,3]   [,4]
[1,] 5.8149 3.2683 2.3721 1.3044
[2,] 3.2683 4.5807 3.3246 1.8281
[3,] 2.3721 3.3246 6.0172 3.3087
[4,] 1.3044 1.8281 3.3087 4.5369
  Standard Deviations: 2.4114 2.1403 2.453 2.13 
Marginal variance covariance matrix
       [,1]   [,2]   [,3]   [,4]
[1,] 5.8149 3.2683 2.3721 1.3044
[2,] 3.2683 4.5807 3.3246 1.8281
[3,] 2.3721 3.3246 6.0172 3.3087
[4,] 1.3044 1.8281 3.3087 4.5369
  Standard Deviations: 2.4114 2.1403 2.453 2.13 </code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-36_89b3ca7e7d7720e15141514156833f61">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Modelings including random effects</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>dental_lme <span class="ot">&lt;-</span> <span class="fu">lme</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender, </span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> id,</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental)</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>dental_lmer <span class="ot">&lt;-</span> <span class="fu">lmer</span>(distance<span class="sc">~</span>age<span class="sc">*</span>gender <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> id), <span class="at">data =</span> dental)</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="co"># same model in lmer and lme, G unstructured</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>dental_lmer_age <span class="ot">&lt;-</span> <span class="fu">lmer</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender <span class="sc">+</span> (age <span class="sc">|</span> id), <span class="at">data =</span> dental)</span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>dental_lme_age <span class="ot">&lt;-</span> <span class="fu">lme</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender, </span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">random =</span> <span class="sc">~</span> age <span class="sc">|</span> id,</span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> dental)</span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a><span class="co"># uncorrelated age and id in random effect, G diag</span></span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>dental_lme_age_diag <span class="ot">&lt;-</span> <span class="fu">lme</span>(distance <span class="sc">~</span> age<span class="sc">*</span>gender,</span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> dental,</span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">random =</span> <span class="fu">list</span>(<span class="at">id =</span> <span class="fu">pdDiag</span>(<span class="at">form =</span> <span class="sc">~</span>age))) <span class="co"># heterogenous, but uncorrelated age and id (age || id) in lmer</span></span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>dental_lmer_age_diag <span class="ot">&lt;-</span> <span class="fu">lmer</span>(distance<span class="sc">~</span>age<span class="sc">*</span>gender <span class="sc">+</span> (age <span class="sc">||</span> id), <span class="at">data =</span> dental)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-37_82b00c9119dddced770e661d1c6591cc">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_lme)</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_lme_age)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_lme_age_diag)</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_lme, <span class="at">type =</span> <span class="st">"marginal"</span>) <span class="co"># same as CS variance</span></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_lme_age, <span class="at">type =</span> <span class="st">"marginal"</span>)</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a><span class="fu">getVarCov</span>(dental_lme_age_diag, <span class="at">type =</span> <span class="st">"marginal"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Random effects variance covariance matrix
            (Intercept)
(Intercept)      3.2986
  Standard Deviations: 1.8162 
Random effects variance covariance matrix
            (Intercept)       age
(Intercept)     5.78640 -0.289630
age            -0.28963  0.032524
  Standard Deviations: 2.4055 0.18035 
Random effects variance covariance matrix
            (Intercept)       age
(Intercept)      2.4168 0.0000000
age              0.0000 0.0077469
  Standard Deviations: 1.5546 0.088017 
id 1 
Marginal variance covariance matrix
       1      2      3      4
1 5.2207 3.2986 3.2986 3.2986
2 3.2986 5.2207 3.2986 3.2986
3 3.2986 3.2986 5.2207 3.2986
4 3.2986 3.2986 3.2986 5.2207
  Standard Deviations: 2.2849 2.2849 2.2849 2.2849 
id 1 
Marginal variance covariance matrix
       1      2      3      4
1 4.9502 3.1751 3.1162 3.0574
2 3.1751 4.9625 3.3176 3.3888
3 3.1162 3.3176 5.2351 3.7202
4 3.0574 3.3888 3.7202 5.7679
  Standard Deviations: 2.2249 2.2277 2.288 2.4016 
id 1 
Marginal variance covariance matrix
       1      2      3      4
1 4.7772 3.0366 3.1605 3.2845
2 3.0366 5.0561 3.3464 3.5014
3 3.1605 3.3464 5.3970 3.7183
4 3.2845 3.5014 3.7183 5.7998
  Standard Deviations: 2.1857 2.2486 2.3231 2.4083 </code></pre>
</div>
</div>
<p>The covariance stuff is harder to calculate from lmer, I only know how to get from manual components, there may be something better that I don’t know about.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-38_270b4c7e7a82349c3ae20e95165f85d1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># G</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(dental_lmer_age)[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)        age
(Intercept)   5.7744874 -0.2886962
age          -0.2886962  0.0324516
attr(,"stddev")
(Intercept)         age 
  2.4030163   0.1801433 
attr(,"correlation")
            (Intercept)        age
(Intercept)   1.0000000 -0.6669087
age          -0.6669087  1.0000000</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Manual calculation of G = Sigma^2 L'L</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">getME</span>(dental_lmer_age, <span class="st">"Lambda"</span>) <span class="co"># Marginal covariance</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span> <span class="fu">getME</span>(dental_lmer_age, <span class="st">"sigma"</span>)</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>(<span class="fu">crossprod</span>(L) <span class="sc">*</span> sig<span class="sc">^</span><span class="dv">2</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2 x 2 sparse Matrix of class "dsCMatrix"
                           
[1,]  5.7889208 -0.01612650
[2,] -0.0161265  0.01801819</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ZGZ' + R</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>Z <span class="ot">&lt;-</span> <span class="fu">getME</span>(dental_lmer_age, <span class="st">"Z"</span>)</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>G <span class="ot">&lt;-</span> <span class="fu">Matrix</span>(<span class="fu">diag</span>(<span class="dv">27</span>)) <span class="sc">%x%</span> <span class="fu">VarCorr</span>(dental_lmer_age)[[<span class="dv">1</span>]] <span class="co"># hadamard</span></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>R <span class="ot">&lt;-</span> <span class="fu">Matrix</span>(<span class="fu">diag</span>(<span class="fu">rep</span>(<span class="fu">sigma</span>(dental_lmer_age)<span class="sc">^</span><span class="dv">2</span>, <span class="dv">108</span>))) <span class="co"># </span></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>varY <span class="ot">&lt;-</span> Z <span class="sc">%*%</span> G <span class="sc">%*%</span> <span class="fu">t</span>(Z) <span class="sc">+</span> R <span class="co"># Var(Y)</span></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>varY[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>4 x 4 sparse Matrix of class "dgCMatrix"
         1        2        3        4
1 4.948875 3.174083 3.115916 3.057749
2 3.174083 4.962347 3.317362 3.389001
3 3.115916 3.317362 5.235433 3.720254
4 3.057749 3.389001 3.720254 5.768131</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-39_31f71726d02391a6178a671592ea01cc">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># both criteria choose the cs matrix, the covariance is quite similar</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">AIC</span>(dental_lm,</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  dental_het,</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>    dental_un,</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>    dental_hun,</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>    dental_cs, </span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>    dental_csh,</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>    dental_ar,</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>    dental_arh,</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>    dental_lme) <span class="sc">%&gt;%</span> </span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">BIC =</span> <span class="fu">BIC</span>(</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>    dental_lm,</span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>    dental_het,</span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>    dental_un,</span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>    dental_hun,</span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>    dental_cs,</span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>    dental_csh,</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>    dental_ar,</span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>    dental_arh,</span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>    dental_lme)<span class="sc">$</span>BIC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in AIC.default(dental_lm, dental_het, dental_un, dental_hun,
dental_cs, : models are not all fitted to the same number of observations</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in BIC.default(dental_lm, dental_het, dental_un, dental_hun,
dental_cs, : models are not all fitted to the same number of observations</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>           df      AIC      BIC
dental_lm   5 488.2418 501.6524
dental_het  8 498.4114 519.5666
dental_un  11 448.1706 477.2589
dental_hun 14 452.5468 489.5683
dental_cs   6 445.7572 461.6236
dental_csh  9 449.9724 473.7719
dental_ar   9 460.7962 484.5957
dental_arh  9 460.7962 484.5957
dental_lme  6 445.7572 461.6236</code></pre>
</div>
</div>
<p>I was also curious how this compares to just calculating the sample covariance. It seems not all that different from the unstructured covariance matrix. They are different, but it seems the residuals being calculated are different. I thought ML estimator should be pretty close to the unstructured estimate.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-40_56046c434214f748ec4712614bb5f3bc">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sample covariance with group centering</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>dental <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(gender, age) <span class="sc">%&gt;%</span> </span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cdist =</span> distance <span class="sc">-</span> <span class="fu">mean</span>(distance, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span>  <span class="co"># group center</span></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(age), <span class="at">names_from =</span> id, <span class="at">values_from =</span> cdist) <span class="sc">%&gt;%</span> <span class="co"># matrix format for cov </span></span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>age) <span class="sc">%&gt;%</span> </span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.matrix</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">cov</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>         [,1]     [,2]     [,3]     [,4]
[1,] 5.207168 2.612325 3.759834 2.605988
[2,] 2.612325 4.023820 2.814576 3.189576
[3,] 3.759834 2.814576 6.207441 3.971864
[4,] 2.605988 3.189576 3.971864 4.793979</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-41_0e3f1c0d3342556d077d52c5a4ecc0a5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># predictions from each of the models</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>dental_yhat <span class="ot">&lt;-</span> dental <span class="sc">%&gt;%</span> <span class="fu">add_column</span>(<span class="at">un =</span> <span class="fu">predict</span>(dental_un),</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">hun =</span> <span class="fu">predict</span>(dental_hun),</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">cs =</span> <span class="fu">predict</span>(dental_cs),</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">csh =</span> <span class="fu">predict</span>(dental_csh),</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">ar =</span> <span class="fu">predict</span>(dental_ar),</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">arh =</span> <span class="fu">predict</span>(dental_arh),</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">het =</span> <span class="fu">predict</span>(dental_het),</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">lme =</span> <span class="fu">predict</span>(dental_lme),</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">fixed =</span> <span class="fu">predict</span>(dental_fixed),</span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">lm =</span> <span class="fu">predict</span>(dental_lm),</span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">lme_age_diag =</span> <span class="fu">predict</span>(dental_lme_age_diag),</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">lme_age =</span> <span class="fu">predict</span>(dental_lme_age))</span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>dental_yhat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(id <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> distance<span class="sc">:</span>lme_age, <span class="at">names_to =</span> <span class="st">"type"</span>,</span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"response"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(age, response, <span class="at">color =</span> type)) <span class="sc">+</span></span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Here we examine what all these models actually predicted. For the marginal model, most of them are being predicted together, which makes sense. The “distance” are the raw values displayed for one individual. We can see that the fixed model (separate fixed effect for id) and total mean model (lm, also called pooled) runs with the other coefficients. As far as estimates go, there are 3 separate groups, and probably some are due to rounding error. I think theoretically, the gls beta estimates should all be the same?</p>
<p>Now we should also be concered about the effects of <code>vcov</code> of the different models</p>
<p>The sandwich estimators here are also worth comparing. All estimators have the form <span class="math inline">X'\hat\Sigma X</span> with different “meat” for <span class="math inline">\hat \Sigma</span>.</p>
<ul>
<li>HC0: Original white estimator
<ul>
<li><span class="math inline">\hat\Sigma = X'\{\hat\varepsilon_i^2\}_dX</span></li>
</ul></li>
<li>HC1: adjust degrees of freedom of HC0
<ul>
<li><span class="math inline">\hat\Sigma = \frac{N}{N-(k+1)}X'\{\hat\varepsilon_i^2\}_dX</span></li>
</ul></li>
<li>HC2: incorporate leverage
<ul>
<li><span class="math inline">\hat\Sigma = X'\left\{\frac{\hat\varepsilon_i^2}{1- h_{ii}}\right\}_dX</span></li>
</ul></li>
<li>HC3: incorporate leverage with differnt weights
<ul>
<li><span class="math inline">\hat\Sigma = X'\left\{\frac{\hat\varepsilon_i^2}{(1- h_{ii})^2}\right\}_dX</span></li>
</ul></li>
<li>HC4: incorporate leverage with differnt weights
<ul>
<li><span class="math inline">\hat\Sigma = X'\left\{\frac{\hat\varepsilon_i^2}{(1- h_{ii})^{\delta_{i}}}\right\}_dX</span></li>
<li><span class="math inline">\delta_i = \min \{4, \frac{Nh_{ii}}{k + 1}\}</span></li>
</ul></li>
</ul>
<p>For Heteroscedastic errors, there are HAC and Feasible Generalized Least Squares:</p>
<ul>
<li></li>
</ul>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-43_ff5e41487b7b613832ed2f485d66ffe8">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># autocorrelated</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="fu">NeweyWest</span>(dental_lm) <span class="co"># Bartlett kernel weights</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)         age    genderM age:genderM
(Intercept)   1.5805513 -0.11935718 -1.5695856  0.11933808
age          -0.1193572  0.01170302  0.1289546 -0.01245172
genderM      -1.5695856  0.12895463  3.9660625 -0.33731829
age:genderM   0.1193381 -0.01245172 -0.3373183  0.03234847</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">NeweyWest</span>(dental_lm, <span class="at">lag =</span> <span class="dv">0</span>) <span class="co">#??</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)         age    genderM age:genderM
(Intercept)   2.3956703 -0.20743588 -2.4248178  0.20948506
age          -0.2074359  0.02111911  0.2205834 -0.02204344
genderM      -2.4248178  0.22058338  5.1489658 -0.46060717
age:genderM   0.2094851 -0.02204344 -0.4606072  0.04503617</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">NeweyWest</span>(dental_lm, <span class="at">lag =</span> <span class="dv">1</span>) <span class="co"># Bartlett kernel weights</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)         age    genderM age:genderM
(Intercept)   2.1562577 -0.17466452 -2.1221197  0.17190552
age          -0.1746645  0.01688017  0.1804904 -0.01725257
genderM      -2.1221197  0.18049043  4.8322769 -0.41689414
age:genderM   0.1719055 -0.01725257 -0.4168941  0.03939289</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vcovHAC</span>(dental_lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)          age     genderM  age:genderM
(Intercept)  0.98793825 -0.076828444 -0.85989312  0.061591713
age         -0.07682844  0.008502217  0.07210153 -0.007497458
genderM     -0.85989312  0.072101528  2.07233695 -0.160873344
age:genderM  0.06159171 -0.007497458 -0.16087334  0.015550636</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vcovHAC</span>(dental_lm, <span class="at">weights =</span> weightsAndrews)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)          age     genderM  age:genderM
(Intercept)  0.98793825 -0.076828444 -0.85989312  0.061591713
age         -0.07682844  0.008502217  0.07210153 -0.007497458
genderM     -0.85989312  0.072101528  2.07233695 -0.160873344
age:genderM  0.06159171 -0.007497458 -0.16087334  0.015550636</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vcovHAC</span>(dental_lm, <span class="at">weights =</span> weightsLumley)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)         age    genderM age:genderM
(Intercept)   2.3180590 -0.19769599 -2.6024990  0.21980484
age          -0.1976960  0.01957897  0.2314731 -0.02220841
genderM      -2.6024990  0.23147308  5.2298357 -0.44622075
age:genderM   0.2198048 -0.02220841 -0.4462207  0.04203649</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-44_eebc0a355b4b2f9c7c265cff96d37d30">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># vcovPL is most appropriate I think, bu</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="fu">vcovPL</span>(dental_lm, <span class="at">cluster =</span> <span class="sc">~</span>age , <span class="at">kernel =</span> <span class="st">"Bartlett"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)         age     genderM age:genderM
(Intercept)  0.36436517 -0.03429299 -0.37025009  0.03404624
age         -0.03429299  0.00491551  0.03440182 -0.00480140
genderM     -0.37025009  0.03440182  1.50047722 -0.12229004
age:genderM  0.03404624 -0.00480140 -0.12229004  0.01217204</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vcovPL</span>(dental_lm, <span class="at">cluster =</span> dental<span class="sc">$</span>agef) <span class="co"># on the order</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)         age     genderM age:genderM
(Intercept)  0.36436517 -0.03429299 -0.37025009  0.03404624
age         -0.03429299  0.00491551  0.03440182 -0.00480140
genderM     -0.37025009  0.03440182  1.50047722 -0.12229004
age:genderM  0.03404624 -0.00480140 -0.12229004  0.01217204</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vcovCL</span>(dental_lm, <span class="at">cluster =</span> <span class="sc">~</span>id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)          age     genderM  age:genderM
(Intercept)  0.56190635 -0.031179559 -0.56190635  0.031179559
age         -0.03117956  0.004258417  0.03117956 -0.004258417
genderM     -0.56190635  0.031179559  2.02816740 -0.145146882
age:genderM  0.03117956 -0.004258417 -0.14514688  0.014592405</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vcov</span>(dental_cs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)          age    genderM  age:genderM
(Intercept)   1.4006890 -0.096102802 -1.4006890  0.096102802
age          -0.0961028  0.008736618  0.0961028 -0.008736618
genderM      -1.4006890  0.096102802  2.3636627 -0.162173479
age:genderM   0.0961028 -0.008736618 -0.1621735  0.014743044</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vcov</span>(dental_lme_age) <span class="co"># is cluster supposed to be age?</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>            (Intercept)        age    genderM age:genderM
(Intercept)   1.5089562 -0.1121399 -1.5089562  0.11213994
age          -0.1121399  0.0107577  0.1121399 -0.01075770
genderM      -1.5089562  0.1121399  2.5463635 -0.18923615
age:genderM   0.1121399 -0.0107577 -0.1892361  0.01815361</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-45_6a7bb7ec4ddb3383e4d4d1862af85d88">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># standard errors are wildly different, why and how? Need to test reliability.</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tidy</span>(dental_csh)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 5
  term        estimate std.error statistic  p.value
  &lt;chr&gt;          &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;
1 (Intercept)   17.4      1.18      14.7   3.08e-27
2 age            0.479    0.0929     5.15  1.22e- 6
3 genderM       -1.27     1.53      -0.831 4.08e- 1
4 age:genderM    0.316    0.121      2.62  1.02e- 2</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coeftest</span>(dental_lm, <span class="at">vcov =</span> vcovCL, <span class="at">cluster =</span> <span class="sc">~</span>id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
t test of coefficients:

             Estimate Std. Error t value  Pr(&gt;|t|)    
(Intercept) 17.372727   0.749604 23.1759 &lt; 2.2e-16 ***
age          0.479545   0.065257  7.3486 4.712e-11 ***
genderM     -1.032102   1.424137 -0.7247   0.47025    
age:genderM  0.304830   0.120799  2.5234   0.01313 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coeftest</span>(dental_lm, <span class="at">vcov =</span> vcovPL, <span class="at">cluster =</span> <span class="sc">~</span>id) <span class="co"># almost 10x lower standard error?</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
t test of coefficients:

             Estimate Std. Error  t value  Pr(&gt;|t|)    
(Intercept) 17.372727   0.056518 307.3812 &lt; 2.2e-16 ***
age          0.479546   0.004701 102.0090 &lt; 2.2e-16 ***
genderM     -1.032102   0.527300  -1.9573   0.05299 .  
age:genderM  0.304829   0.043791   6.9610 3.128e-10 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coeftest</span>(dental_lm, <span class="at">vcov =</span> vcovHAC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
t test of coefficients:

             Estimate Std. Error t value  Pr(&gt;|t|)    
(Intercept) 17.372727   0.993951 17.4785 &lt; 2.2e-16 ***
age          0.479545   0.092207  5.2007 9.998e-07 ***
genderM     -1.032102   1.439561 -0.7170   0.47501    
age:genderM  0.304830   0.124702  2.4445   0.01619 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Introducing this many different ways for estimating “robust” standard errors is quite confusing.</p>
</section>
</section>
</section>
<section id="random-coefficient-models" class="level1" data-number="20">
<h1 data-number="20"><span class="header-section-number">20</span> Random Coefficient Models</h1>
<p>This is a method of adjusting for differing slopes and effects.</p>
<section id="example-autism" class="level2" data-number="20.1">
<h2 data-number="20.1" class="anchored" data-anchor-id="example-autism"><span class="header-section-number">20.1</span> Example: Autism</h2>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-46_c8caade76d9598f825c34aaa94367b79">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>autism <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/autism.csv"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sicdegp =</span> <span class="fu">factor</span>(sicdegp),</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">childid =</span> <span class="fu">factor</span>(childid)) </span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>autism_copy <span class="ot">&lt;-</span> autism</span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a>autism <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(age, vsae, <span class="at">group =</span> childid, <span class="at">color =</span> sicdegp)) <span class="sc">+</span></span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>sicdegp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row(s) containing missing values (geom_path).</code></pre>
</div>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-47_2d96e65d0025538b083eadefad3f1221">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># how many distinct in each</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>autism <span class="sc">%&gt;%</span> <span class="fu">count</span>(sicdegp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  sicdegp   n
1       1 192
2       2 255
3       3 165</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>autism <span class="sc">%&gt;%</span> <span class="fu">count</span>(sicdegp, childid) <span class="sc">%&gt;%</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> sicdegp, <span class="at">values_from =</span> n)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 158 × 4
   childid   `1`   `2`   `3`
   &lt;fct&gt;   &lt;int&gt; &lt;int&gt; &lt;int&gt;
 1 2           5    NA    NA
 2 6           4    NA    NA
 3 8           3    NA    NA
 4 10          4    NA    NA
 5 13          3    NA    NA
 6 22          3    NA    NA
 7 31          4    NA    NA
 8 32          3    NA    NA
 9 38          5    NA    NA
10 41          3    NA    NA
# … with 148 more rows
# ℹ Use `print(n = ...)` to see more rows</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 missing vsae values</span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>autism <span class="sc">%&gt;%</span> <span class="fu">lapply</span>(rlang<span class="sc">::</span><span class="fu">as_function</span>(<span class="sc">~</span><span class="fu">which</span>(<span class="fu">is.na</span>(.x)))) <span class="co"># columnwise which missing</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$age
integer(0)

$vsae
[1] 507 553

$sicdegp
integer(0)

$childid
integer(0)</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-48_2b64934894e0f0852d177d3e31904a33">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="co"># without child id</span></span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a>autism_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(vsae <span class="sc">~</span> age<span class="sc">*</span>sicdegp, <span class="at">data =</span> autism) <span class="co"># autism</span></span>
<span id="cb128-3"><a href="#cb128-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-4"><a href="#cb128-4" aria-hidden="true" tabindex="-1"></a><span class="co"># with child id</span></span>
<span id="cb128-5"><a href="#cb128-5" aria-hidden="true" tabindex="-1"></a>autism_lm_id <span class="ot">&lt;-</span> <span class="fu">lm</span>(vsae <span class="sc">~</span> age<span class="sc">*</span>sicdegp <span class="sc">+</span> childid, <span class="at">data =</span> autism)</span>
<span id="cb128-6"><a href="#cb128-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-7"><a href="#cb128-7" aria-hidden="true" tabindex="-1"></a><span class="co"># with id varying slopes</span></span>
<span id="cb128-8"><a href="#cb128-8" aria-hidden="true" tabindex="-1"></a>autism_lm_id_slope <span class="ot">&lt;-</span> <span class="fu">lm</span>(vsae <span class="sc">~</span> age<span class="sc">*</span>sicdegp <span class="sc">+</span> childid <span class="sc">+</span> childid<span class="sc">:</span>age <span class="sc">+</span> childid<span class="sc">:</span>sicdegp, <span class="at">data =</span> autism)</span>
<span id="cb128-9"><a href="#cb128-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-10"><a href="#cb128-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-11"><a href="#cb128-11" aria-hidden="true" tabindex="-1"></a><span class="co"># predictions, plotted raw</span></span>
<span id="cb128-12"><a href="#cb128-12" aria-hidden="true" tabindex="-1"></a>autism_predict <span class="ot">&lt;-</span> autism <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">complete.cases</span>(.)) <span class="sc">%&gt;%</span> </span>
<span id="cb128-13"><a href="#cb128-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(</span>
<span id="cb128-14"><a href="#cb128-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">yhat_lm =</span> <span class="fu">predict</span>(autism_lm),</span>
<span id="cb128-15"><a href="#cb128-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">yhat_lm_id =</span> <span class="fu">predict</span>(autism_lm_id),</span>
<span id="cb128-16"><a href="#cb128-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">yhat_lm_id_slope =</span> <span class="fu">predict</span>(autism_lm_id_slope))</span>
<span id="cb128-17"><a href="#cb128-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-18"><a href="#cb128-18" aria-hidden="true" tabindex="-1"></a>autism_predict <span class="sc">%&gt;%</span> <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(vsae, <span class="fu">starts_with</span>(<span class="st">"yhat"</span>)),</span>
<span id="cb128-19"><a href="#cb128-19" aria-hidden="true" tabindex="-1"></a>                                <span class="at">names_to =</span> <span class="st">"type"</span>,</span>
<span id="cb128-20"><a href="#cb128-20" aria-hidden="true" tabindex="-1"></a>                                <span class="at">values_to =</span> <span class="st">"y"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb128-21"><a href="#cb128-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(age, y, <span class="at">group =</span> childid, <span class="at">color =</span> type)) <span class="sc">+</span></span>
<span id="cb128-22"><a href="#cb128-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">alpha =</span> .<span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb128-23"><a href="#cb128-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(type<span class="sc">~</span>sicdegp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-49_3334f85ac037383cdec06f9a12273092">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fully fixed</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">slope1 =</span> <span class="fu">coef</span>(autism_lm)[<span class="dv">2</span>],</span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">slope2 =</span> <span class="fu">sum</span>(<span class="fu">coef</span>(autism_lm)[<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">5</span>)]),</span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">slope3 =</span> <span class="fu">sum</span>(<span class="fu">coef</span>(autism_lm)[<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>slope1.age     slope2     slope3 
  2.615931   4.233238   6.980310 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="co"># with id</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">slope1 =</span> <span class="fu">coef</span>(autism_lm_id)[<span class="st">"age"</span>],</span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">slope2 =</span> <span class="fu">sum</span>(<span class="fu">coef</span>(autism_lm_id)[<span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"age:sicdegp2"</span>)]),</span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">slope3 =</span> <span class="fu">sum</span>(<span class="fu">coef</span>(autism_lm_id)[<span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"age:sicdegp3"</span>)]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>slope1.age     slope2     slope3 
  2.598931   4.072511   7.075103 </code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-50_467b64cebedd9851c93ab544e33acc6f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: with id_slope</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a><span class="co"># average coefficients for each fitted child? seems okay way to summarize this fixed effects model.</span></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>rg_lm_id_slope <span class="ot">&lt;-</span> <span class="fu">ref_grid</span>(autism_lm_id_slope, <span class="at">at =</span> <span class="fu">list</span>(<span class="st">"age"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>), <span class="st">"sicdegp"</span>)) <span class="co"># nesting</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>NOTE: A nesting structure was detected in the fitted model:
    childid %in% sicdegp</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>emmp_lm_id_slope <span class="ot">&lt;-</span> <span class="fu">pairs</span>(<span class="fu">emmeans</span>(rg_lm_id_slope, <span class="at">specs =</span> <span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"sicdegp"</span>), <span class="at">by =</span> <span class="st">"childid"</span>)) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>emmp_lm_id_slope <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sicdegp) <span class="sc">%&gt;%</span> </span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_slope =</span> <span class="sc">-</span><span class="fu">mean</span>(estimate, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(mean_slope)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3.004215 3.631826 7.258098</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-51_e079e6b8a28284681549d4a38cd9be0a">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co"># random models</span></span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>autism_lme_id <span class="ot">&lt;-</span> <span class="fu">lme</span>(vsae<span class="sc">~</span>age<span class="sc">*</span>sicdegp,</span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> childid,</span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> autism[<span class="fu">complete.cases</span>(autism),])</span>
<span id="cb137-5"><a href="#cb137-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-6"><a href="#cb137-6" aria-hidden="true" tabindex="-1"></a>autism_lmer_id <span class="ot">&lt;-</span> <span class="fu">lmer</span>(vsae<span class="sc">~</span> age <span class="sc">*</span> sicdegp <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span> childid),</span>
<span id="cb137-7"><a href="#cb137-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> autism)</span>
<span id="cb137-8"><a href="#cb137-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-9"><a href="#cb137-9" aria-hidden="true" tabindex="-1"></a>autism_lmer_id_slope <span class="ot">&lt;-</span> <span class="fu">lmer</span>(vsae<span class="sc">~</span> age <span class="sc">*</span> sicdegp <span class="sc">+</span> ( age <span class="sc">|</span> childid),</span>
<span id="cb137-10"><a href="#cb137-10" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data =</span> autism)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>autism_lmer_id_slope_split <span class="ot">&lt;-</span> <span class="fu">lmer</span>(vsae<span class="sc">~</span> age <span class="sc">*</span> sicdegp <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> childid) <span class="sc">+</span> ( <span class="dv">0</span> <span class="sc">+</span> age <span class="sc">|</span> childid),</span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data =</span> autism)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="co"># uncorrelated intercept and age</span></span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a>autism_lmer_id_slope_uncor <span class="ot">&lt;-</span> <span class="fu">lmer</span>(vsae <span class="sc">~</span> age <span class="sc">*</span> sicdegp <span class="sc">+</span> (age <span class="sc">||</span> childid),</span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data =</span> autism)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>boundary (singular) fit: see help('isSingular')</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="co"># unstructured covariance matrix</span></span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a><span class="co"># gls(vsae~age*sicdegp,</span></span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     random = ~1 | childid,</span></span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     correlation = corSymm(form = ~1|sicdegp),</span></span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     weights = varIdent(form = ~ age)) </span></span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-7"><a href="#cb143-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-8"><a href="#cb143-8" aria-hidden="true" tabindex="-1"></a>autism_lme_id_slope <span class="ot">&lt;-</span> <span class="fu">lme</span>(vsae<span class="sc">~</span>age<span class="sc">*</span>sicdegp,</span>
<span id="cb143-9"><a href="#cb143-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">random =</span> <span class="sc">~</span> age <span class="sc">|</span> childid,</span>
<span id="cb143-10"><a href="#cb143-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">data =</span> autism[<span class="fu">complete.cases</span>(autism),],</span>
<span id="cb143-11"><a href="#cb143-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">control =</span> <span class="fu">lmeControl</span>(<span class="at">opt =</span> <span class="st">"optim"</span>))</span>
<span id="cb143-12"><a href="#cb143-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-13"><a href="#cb143-13" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(autism_lme_id_slope)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed-effects model fit by REML
  Data: autism[complete.cases(autism), ] 
      AIC      BIC    logLik
  4716.13 4760.166 -2348.065

Random effects:
 Formula: ~age | childid
 Structure: General positive-definite, Log-Cholesky parametrization
            StdDev   Corr  
(Intercept) 8.602952 (Intr)
age         4.019888 -0.998
Residual    7.769026       

Fixed effects:  vsae ~ age * sicdegp 
                 Value Std.Error  DF   t-value p-value
(Intercept)   1.843174 1.6899783 449  1.090650  0.2760
age           2.971975 0.6268443 449  4.741170  0.0000
sicdegp2     -0.324782 2.2318720 155 -0.145520  0.8845
sicdegp3     -3.858107 2.4775418 155 -1.557232  0.1215
age:sicdegp2  0.715160 0.8290366 449  0.862640  0.3888
age:sicdegp3  4.334815 0.9181896 449  4.721046  0.0000
 Correlation: 
             (Intr) age    scdgp2 scdgp3 ag:sc2
age          -0.891                            
sicdegp2     -0.757  0.675                     
sicdegp3     -0.682  0.608  0.517              
age:sicdegp2  0.674 -0.756 -0.891 -0.460       
age:sicdegp3  0.609 -0.683 -0.461 -0.888  0.516

Standardized Within-Group Residuals:
        Min          Q1         Med          Q3         Max 
-3.87638128 -0.34680674 -0.02510791  0.31120771  5.28293064 

Number of Observations: 610
Number of Groups: 158 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lmer_id</span></span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="fu">fixef</span>(autism_lmer_id)[<span class="st">"age"</span>],</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sum</span>(<span class="fu">fixef</span>(autism_lmer_id)[<span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"age:sicdegp2"</span>)]),</span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sum</span>(<span class="fu">fixef</span>(autism_lmer_id)[<span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"age:sicdegp3"</span>)]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     age                   
2.618120 4.147079 7.041573 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lme_id_slope</span></span>
<span id="cb147-2"><a href="#cb147-2" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="fu">fixef</span>(autism_lme_id_slope)[<span class="st">"age"</span>],</span>
<span id="cb147-3"><a href="#cb147-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sum</span>(<span class="fu">fixef</span>(autism_lme_id_slope)[<span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"age:sicdegp2"</span>)]),</span>
<span id="cb147-4"><a href="#cb147-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sum</span>(<span class="fu">fixef</span>(autism_lme_id_slope)[<span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"age:sicdegp3"</span>)]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     age                   
2.971975 3.687136 7.306791 </code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-52_aa220141626690824a44cd1273d3a657">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="co"># random models</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="baseline-adjustment" class="level1" data-number="21">
<h1 data-number="21"><span class="header-section-number">21</span> Baseline adjustment</h1>
<p>There are 4 ways that we can discuss for baseline adjustment from <span class="citation" data-cites="fitzmaurice_applied_2011">Fitzmaurice, Laird, and Ware (<a href="#ref-fitzmaurice_applied_2011" role="doc-biblioref">2011</a>)</span> chapter 5.7</p>
<ol type="1">
<li>Keep in outcome vector, no assumptions about group differences in mean response at baseline. “baseline group different” (basically what we did prior)</li>
<li>Keep in outcome vecto, make assumption that group means are equal at baseline (like in randomized trial) “no baseline</li>
<li>Subtract baseline response from post-baseline. Analyze the diffrences from baseline</li>
<li>Use baseline value as a covariates in the analysis of the post-baseline responses.</li>
</ol>
<p>In summary, Fitzmaurice shows and recommends:</p>
<ul>
<li>1 = 3.</li>
<li>2 is just as efficient as 4</li>
<li>2 is more efficient than 1.</li>
<li>assumption 2 is most accurate in a randomized control trial, in which you expect baseline to be similar. introducing backdoor bias (conditioning on collider) may happen in observational studies. In which case use 1</li>
</ul>
<p>Depending on the strategy that you choose, the interpretation of the coefficients will differ, so beware, although some cases reduce to linear combinations of the coefficients.</p>
<p>We examine the differences of these 4 approaches on the TLC dataset.</p>
<section id="method-implementations" class="level2 tabset" data-number="21.1">
<h2 class="tabset anchored" data-number="21.1" data-anchor-id="method-implementations"><span class="header-section-number">21.1</span> Method Implementations</h2>
<section id="raw" class="level3" data-number="21.1.1">
<h3 data-number="21.1.1" class="anchored" data-anchor-id="raw"><span class="header-section-number">21.1.1</span> 1. raw</h3>
<figure class="figure">
<center>
<p><img src="img/tlc_baseline_raw.png" style="width:50%" class="figure-img"></p>
<figcaption class="figure-caption">
</figcaption>
</center>
</figure>
<p>This method is just the standard response profile analysis, which we’ve shown above.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-53_b007fa7944744c9001a423ae702aedaf">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a>tlc_baseline_1 <span class="ot">&lt;-</span> tlc_gls</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="constrain-baseline" class="level3" data-number="21.1.2">
<h3 data-number="21.1.2" class="anchored" data-anchor-id="constrain-baseline"><span class="header-section-number">21.1.2</span> 2. constrain baseline</h3>
<figure class="figure">
<center>
<p><img src="img/tlc_constrain_baseline.png" style="width:80%" class="figure-img"></p>
<figcaption class="figure-caption">
</figcaption>
</center>
</figure>
<p>In order to force baseline to be modeled the same, we need to remove the main effect for treament, and the interaction term involving the baseline. Unfortunately we can’t remove a single level of the interaction if we specify <code>week*trt</code>, nor does <code>gls</code> accept a manually made model matrix, so we must construct the variables through the formula interface.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-54_e6428e6f348b7f3e79077fb4a8b2763b">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep in outcome, assume similar intercept </span></span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="co"># remove main effect, and week0 interactions.</span></span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a><span class="co"># gls doesn't take model.matrix, so manual specification through formula is the way to do it</span></span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a><span class="co"># update also acts kinda weird</span></span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a><span class="co"># table 5.7  in Fitzmaurice</span></span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a><span class="co"># https://content.sph.harvard.edu/fitzmaur/ala2e/</span></span>
<span id="cb151-7"><a href="#cb151-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-8"><a href="#cb151-8" aria-hidden="true" tabindex="-1"></a>tlc_baseline_2 <span class="ot">&lt;-</span> <span class="fu">gls</span>(lead <span class="sc">~</span> week <span class="sc">+</span> </span>
<span id="cb151-9"><a href="#cb151-9" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">I</span>(week <span class="sc">==</span> <span class="st">"w1"</span> <span class="sc">&amp;</span> trt <span class="sc">==</span> <span class="st">"A"</span>) <span class="sc">+</span></span>
<span id="cb151-10"><a href="#cb151-10" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">I</span>(week <span class="sc">==</span> <span class="st">"w4"</span> <span class="sc">&amp;</span> trt <span class="sc">==</span> <span class="st">"A"</span>) <span class="sc">+</span></span>
<span id="cb151-11"><a href="#cb151-11" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">I</span>(week <span class="sc">==</span> <span class="st">"w6"</span> <span class="sc">&amp;</span> trt <span class="sc">==</span> <span class="st">"A"</span>),</span>
<span id="cb151-12"><a href="#cb151-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> tlc,</span>
<span id="cb151-13"><a href="#cb151-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> time <span class="sc">|</span> id),</span>
<span id="cb151-14"><a href="#cb151-14" aria-hidden="true" tabindex="-1"></a>                      <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> week))</span>
<span id="cb151-15"><a href="#cb151-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-16"><a href="#cb151-16" aria-hidden="true" tabindex="-1"></a><span class="co"># type III fixed effect test for interaction hypothesis</span></span>
<span id="cb151-17"><a href="#cb151-17" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(tlc_baseline_2, <span class="at">Terms =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>) <span class="co"># Chisq = F * ndf = 111.94</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Denom. DF: 393 
 F-test for: I(week == "w1" &amp; trt == "A"), I(week == "w4" &amp; trt == "A"), I(week == "w6" &amp; trt == "A") 
  numDF  F-value p-value
1     3 37.31422  &lt;.0001</code></pre>
</div>
</div>
</section>
<section id="change-from-baseline" class="level3" data-number="21.1.3">
<h3 data-number="21.1.3" class="anchored" data-anchor-id="change-from-baseline"><span class="header-section-number">21.1.3</span> 3. change from baseline</h3>
<figure class="figure">
<center>
<p><img src="img/tlc_baseline_change.png" style="width:60%" class="figure-img"></p>
<figcaption class="figure-caption">
</figcaption>
</center>
</figure>
<p>This analysis is just a reframed version of the first analysis, at least in terms of estimates.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-55_0b1e37877b09bd1ece6cedda157142a7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co"># modeling the difference</span></span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a><span class="co"># time - 1, because gls throws error. corSymm needs consecutive integers starting at 1.</span></span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>tlc_baseline_3 <span class="ot">&lt;-</span> <span class="fu">gls</span>(lead_diff <span class="sc">~</span> trt<span class="sc">*</span>week,</span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> time<span class="dv">-1</span> <span class="sc">|</span> id), <span class="co"># unstructured</span></span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> week),  <span class="co"># heterogenous</span></span>
<span id="cb153-6"><a href="#cb153-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> tlc_diff <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">trt =</span> <span class="fu">factor</span>(trt, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"P"</span>, <span class="st">"A"</span>))))</span>
<span id="cb153-7"><a href="#cb153-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-8"><a href="#cb153-8" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(tlc_baseline_3, <span class="at">Terms =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">4</span>)) <span class="co"># trt x week interaction test is now test on combined c(trt, trt:week) of diff</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Denom. DF: 294 
 F-test for: trt, trt:week 
  numDF  F-value p-value
1     3 35.92872  &lt;.0001</code></pre>
</div>
</div>
<div style="display:flex;justify-content:center;align-items:flex-start">
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-56_027789d6c9f7a8c1e282ad54c486a26d">
<div class="cell-output-display">

<table style="width:30%; float: left; margin-right: 10px;" class="table">
<caption>Baseline Change Coefficients</caption>
 <thead>
  <tr>
   <th style="text-align:left;"> term </th>
   <th style="text-align:right;"> change </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> (Intercept) </td>
   <td style="text-align:right;"> -1.612 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA </td>
   <td style="text-align:right;"> -11.406 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> weekw4 </td>
   <td style="text-align:right;"> -0.590 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> weekw6 </td>
   <td style="text-align:right;"> -1.014 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA:weekw4 </td>
   <td style="text-align:right;"> 2.582 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA:weekw6 </td>
   <td style="text-align:right;"> 8.254 </td>
  </tr>
</tbody>
</table>

</div>
<div class="cell-output-display">

<table style="width:30%; " class="table">
<caption>Raw Coefficients</caption>
 <thead>
  <tr>
   <th style="text-align:left;"> term </th>
   <th style="text-align:right;"> raw </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> (Intercept) </td>
   <td style="text-align:right;"> 26.272 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA </td>
   <td style="text-align:right;"> 0.268 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> weekw1 </td>
   <td style="text-align:right;"> -1.612 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> weekw4 </td>
   <td style="text-align:right;"> -2.202 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> weekw6 </td>
   <td style="text-align:right;"> -2.626 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA:weekw1 </td>
   <td style="text-align:right;"> -11.406 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA:weekw4 </td>
   <td style="text-align:right;"> -8.824 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA:weekw6 </td>
   <td style="text-align:right;"> -3.152 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>To see that they are the same, notice that the baseline change model term estimates are simply linear combinations of the raw model.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-57_1d6b2239c2760038aaf856318f6e7685">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tribble</span>(<span class="sc">~</span><span class="st">`</span><span class="at">change model</span><span class="st">`</span>, <span class="sc">~</span><span class="st">`</span><span class="at">raw model terms</span><span class="st">`</span>, <span class="sc">~</span><span class="st">`</span><span class="at">value</span><span class="st">`</span> , <span class="sc">~</span><span class="st">`</span><span class="at">interpretation</span><span class="st">`</span>,</span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>        <span class="st">"(Intercept)"</span>, <span class="st">"weekw1"</span>, <span class="st">"-1.612"</span>,  <span class="st">"in placebo, diff between w1 and baseline"</span>,</span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">"trtA"</span>, <span class="st">"trtA:weekw1"</span>, <span class="st">"-11.406"</span>,  <span class="st">"difference in trt slopes from baseline to w1"</span>,</span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"weekw4"</span>, <span class="st">"weekw4 - weekw1"</span>,<span class="st">"-0.59"</span>, <span class="st">"in placebo, diff between w4 and w1"</span>,</span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"trtA:weekw4"</span>, <span class="st">"trtA:weekw4 - trtA:weekw1"</span>,<span class="st">"2.582"</span>, <span class="st">"diff in trt slope from w4 to w1"</span>) <span class="sc">%&gt;%</span> <span class="fu">kbl</span>() <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
 <thead>
  <tr>
   <th style="text-align:left;"> change model </th>
   <th style="text-align:left;"> raw model terms </th>
   <th style="text-align:left;"> value </th>
   <th style="text-align:left;"> interpretation </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> (Intercept) </td>
   <td style="text-align:left;"> weekw1 </td>
   <td style="text-align:left;"> -1.612 </td>
   <td style="text-align:left;"> in placebo, diff between w1 and baseline </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA </td>
   <td style="text-align:left;"> trtA:weekw1 </td>
   <td style="text-align:left;"> -11.406 </td>
   <td style="text-align:left;"> difference in trt slopes from baseline to w1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> weekw4 </td>
   <td style="text-align:left;"> weekw4 - weekw1 </td>
   <td style="text-align:left;"> -0.59 </td>
   <td style="text-align:left;"> in placebo, diff between w4 and w1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> trtA:weekw4 </td>
   <td style="text-align:left;"> trtA:weekw4 - trtA:weekw1 </td>
   <td style="text-align:left;"> 2.582 </td>
   <td style="text-align:left;"> diff in trt slope from w4 to w1 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p>Given that they are the same (or linear combinations), what about the standard errors of the estimates? Let’s check for “weekw4”</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-58_93733c2a8ca69f0be74f465cea410c77">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>) <span class="sc">%*%</span> tlc_gls<span class="sc">$</span>varBeta[<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>, <span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>] <span class="sc">%*%</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="co"># old model std err</span></span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a>      tlc_baseline_3<span class="sc">$</span>varBeta[<span class="st">"weekw4"</span>, <span class="st">"weekw4"</span>]) <span class="co"># baseline change model std err</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>          [,1]     [,2]
[1,] 0.4130701 0.413059</code></pre>
</div>
</div>
<p>They are the same!</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-59_ced0cead5df221d44502de6a24e1a427">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">list</span>(<span class="at">raw =</span> tlc_gls<span class="sc">$</span>varBeta,</span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">base3=</span>tlc_baseline_3<span class="sc">$</span>varBeta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$raw
            (Intercept)        trtA     weekw1     weekw4      weekw6
(Intercept)  0.50451606 -0.50451606 -0.1223674 -0.1105218 -0.06048317
trtA        -0.50451606  1.00903212  0.1223674  0.1105218  0.06048317
weekw1      -0.12236737  0.12236737  0.6271371  0.4390662  0.27183933
weekw4      -0.11052175  0.11052175  0.4390662  0.6640654  0.27889731
weekw6      -0.06048317  0.06048317  0.2718393  0.2788973  0.78947714
trtA:weekw1  0.12236737 -0.24473475 -0.6271371 -0.4390662 -0.27183933
trtA:weekw4  0.11052175 -0.22104350 -0.4390662 -0.6640654 -0.27889731
trtA:weekw6  0.06048317 -0.12096634 -0.2718393 -0.2788973 -0.78947714
            trtA:weekw1 trtA:weekw4 trtA:weekw6
(Intercept)   0.1223674   0.1105218  0.06048317
trtA         -0.2447347  -0.2210435 -0.12096634
weekw1       -0.6271371  -0.4390662 -0.27183933
weekw4       -0.4390662  -0.6640654 -0.27889731
weekw6       -0.2718393  -0.2788973 -0.78947714
trtA:weekw1   1.2542741   0.8781324  0.54367866
trtA:weekw4   0.8781324   1.3281308  0.55779461
trtA:weekw6   0.5436787   0.5577946  1.57895428

$base3
            (Intercept)       trtA     weekw4     weekw6 trtA:weekw4
(Intercept)   0.6271416 -0.6271416 -0.1880546 -0.3553027   0.1880546
trtA         -0.6271416  1.2542831  0.1880546  0.3553027  -0.3761093
weekw4       -0.1880546  0.1880546  0.4130590  0.1951241  -0.4130590
weekw6       -0.3553027  0.3553027  0.1951241  0.8729404  -0.1951241
trtA:weekw4   0.1880546 -0.3761093 -0.4130590 -0.1951241   0.8261179
trtA:weekw6   0.3553027 -0.7106053 -0.1951241 -0.8729404   0.3902482
            trtA:weekw6
(Intercept)   0.3553027
trtA         -0.7106053
weekw4       -0.1951241
weekw6       -0.8729404
trtA:weekw4   0.3902482
trtA:weekw6   1.7458807</code></pre>
</div>
</div>
<p>The marginal covariances are also the same <em>if we change the estimation method to ML</em>.</p>
<p>The idea here was that we could calculate the change from baseline covariance estimate from the raw model estimates of covariance.</p>
<p><span class="math display">
\begin{aligned}
\underbrace{\widehat{\mathrm{Cov}}(Y_{i2} -Y_{i1}, Y_{i2} -Y_{i1})}_{\Delta \text{ baseline model estimates}} = \underbrace{\widehat{\mathrm{Var}}(Y_{{i2}}) - 2\widehat{\mathrm{Cov}}(Y_{i2}, Y_{i1}) + \widehat{\mathrm{Var}}(Y_{i1})}_{\text{Raw model estimates}}
\end{aligned}
</span></p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-60_20f76c78a153f179dab81064389dea72">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="co"># We note this is NOT true when we use the REML estimates</span></span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a><span class="co"># linear </span></span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>,</span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a>              <span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>,</span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a>              <span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="at">byrow =</span> <span class="cn">TRUE</span>, <span class="at">ncol =</span> <span class="dv">4</span>)</span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a><span class="co"># not the same!</span></span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a><span class="fu">list</span>(<span class="st">`</span><span class="at">from_baseline_change_model</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">getVarCov</span>(tlc_baseline_3),</span>
<span id="cb160-9"><a href="#cb160-9" aria-hidden="true" tabindex="-1"></a>     <span class="st">`</span><span class="at">from_raw_model</span><span class="st">`</span> <span class="ot">=</span>  L <span class="sc">%*%</span> <span class="fu">getVarCov</span>(tlc_baseline_1) <span class="sc">%*%</span> <span class="fu">t</span>(L))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$from_baseline_change_model
Marginal variance covariance matrix
       [,1]   [,2]   [,3]
[1,] 31.357 21.954 13.592
[2,] 21.954 33.205 13.945
[3,] 13.592 13.945 39.474
  Standard Deviations: 5.5997 5.7623 6.2828 

$from_raw_model
         [,1]     [,2]     [,3]
[1,] 31.35685 21.95331 13.59197
[2,] 21.95331 33.20327 13.94487
[3,] 13.59197 13.94487 39.47386</code></pre>
</div>
</div>
<p>With REML estimates, they are NOT equal. But if we refit the models with ML,</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-61_b0131af57b6004f31621bcd0b404e5c4">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="co"># what if we try ML?</span></span>
<span id="cb162-2"><a href="#cb162-2" aria-hidden="true" tabindex="-1"></a>tlc_baseline_3_ml <span class="ot">&lt;-</span> tlc_baseline_3 <span class="sc">%&gt;%</span> <span class="fu">update</span>(<span class="at">method =</span> <span class="st">"ML"</span>)</span>
<span id="cb162-3"><a href="#cb162-3" aria-hidden="true" tabindex="-1"></a>tlc_baseline_1_ml <span class="ot">&lt;-</span> tlc_gls <span class="sc">%&gt;%</span> <span class="fu">update</span>(<span class="at">method =</span> <span class="st">"ML"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-62_7a2288125ebbdcfde059a8dd62bfbf60">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="fu">list</span>(<span class="st">`</span><span class="at">from_baseline_change_model</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">getVarCov</span>(tlc_baseline_3_ml),</span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>     <span class="st">`</span><span class="at">from_raw_model</span><span class="st">`</span> <span class="ot">=</span>  L <span class="sc">%*%</span> <span class="fu">getVarCov</span>(tlc_baseline_1_ml) <span class="sc">%*%</span> <span class="fu">t</span>(L))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$from_baseline_change_model
Marginal variance covariance matrix
       [,1]   [,2]   [,3]
[1,] 30.729 21.515 13.319
[2,] 21.515 32.541 13.666
[3,] 13.319 13.666 38.684
  Standard Deviations: 5.5434 5.7044 6.2196 

$from_raw_model
         [,1]     [,2]     [,3]
[1,] 30.72886 21.51453 13.31984
[2,] 21.51453 32.54042 13.66637
[3,] 13.31984 13.66637 38.68419</code></pre>
</div>
</div>
</section>
<section id="ancova-change" class="level3" data-number="21.1.4">
<h3 data-number="21.1.4" class="anchored" data-anchor-id="ancova-change"><span class="header-section-number">21.1.4</span> 4. ancova change</h3>
<p>This method results in standard error estimates that are more efficient, but generally only recommended for randomized studies. In observational studies, there’s danger in introducing bias.</p>
<p>We can do ancova on the raw values, or change from baseline values. That is, we could consider either:</p>
<p>Both of these will give the same estimates (maybe obviously when written out this way), but <span class="math inline">\gamma' = \gamma + 1</span>.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-63_0deef1167666fbdc83e610262df444ab">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ancova</span></span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a>tlc_baseline_4 <span class="ot">&lt;-</span> <span class="fu">gls</span>(lead <span class="sc">~</span> trt<span class="sc">*</span>week <span class="sc">+</span> lead_base,</span>
<span id="cb165-3"><a href="#cb165-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> time<span class="dv">-1</span> <span class="sc">|</span> id), <span class="co"># unstructured</span></span>
<span id="cb165-4"><a href="#cb165-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> week),  <span class="co"># heterogeneous</span></span>
<span id="cb165-5"><a href="#cb165-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">data =</span> tlc_diff)</span>
<span id="cb165-6"><a href="#cb165-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-7"><a href="#cb165-7" aria-hidden="true" tabindex="-1"></a>tlc_baseline_4b <span class="ot">&lt;-</span> <span class="fu">gls</span>(lead_diff <span class="sc">~</span> trt<span class="sc">*</span>week <span class="sc">+</span> lead_base,</span>
<span id="cb165-8"><a href="#cb165-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">correlation =</span> <span class="fu">corSymm</span>(<span class="at">form =</span> <span class="sc">~</span> time<span class="dv">-1</span> <span class="sc">|</span> id), <span class="co"># unstructured</span></span>
<span id="cb165-9"><a href="#cb165-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">weights =</span> <span class="fu">varIdent</span>(<span class="at">form =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> week),  <span class="co"># heterogeneous</span></span>
<span id="cb165-10"><a href="#cb165-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> tlc_diff)</span>
<span id="cb165-11"><a href="#cb165-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-12"><a href="#cb165-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">coef</span>(tlc_baseline_4),</span>
<span id="cb165-13"><a href="#cb165-13" aria-hidden="true" tabindex="-1"></a>               <span class="fu">coef</span>(tlc_baseline_4b)) <span class="sc">%&gt;%</span> </span>
<span id="cb165-14"><a href="#cb165-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">colnames&lt;-</span><span class="st">`</span>(<span class="fu">c</span>(<span class="st">"Ancova"</span>, <span class="st">"Ancova Diff"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb165-15"><a href="#cb165-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb165-16"><a href="#cb165-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"term"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>         term      Ancova Ancova Diff
1 (Intercept)   3.5236964   3.5236964
2        trtA -11.3536109 -11.3536109
3      weekw4  -0.5900000  -0.5900000
4      weekw6  -1.0140000  -1.0140000
5   lead_base   0.8045183  -0.1954817
6 trtA:weekw4   2.5820000   2.5820000
7 trtA:weekw6   8.2540000   8.2540000</code></pre>
</div>
</div>
</section>
</section>
<section id="comparison" class="level2" data-number="21.2">
<h2 data-number="21.2" class="anchored" data-anchor-id="comparison"><span class="header-section-number">21.2</span> Comparison</h2>
<p>It’s hard to compare the estimates of the coefficients directly, so we compare them for the hypothesis test of the treatment by time interaction.</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-64_31d792cfc1d967f424176920b31b7853">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="co"># contrast matrix for tlc_baseline_3</span></span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a><span class="co"># trtA (interaction w1), trtA + trtA:w4, trtA + trtA:w6</span></span>
<span id="cb167-3"><a href="#cb167-3" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>,</span>
<span id="cb167-4"><a href="#cb167-4" aria-hidden="true" tabindex="-1"></a>              <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>,</span>
<span id="cb167-5"><a href="#cb167-5" aria-hidden="true" tabindex="-1"></a>              <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="at">byrow =</span> T, <span class="at">nrow =</span> <span class="dv">3</span>)</span>
<span id="cb167-6"><a href="#cb167-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb167-7"><a href="#cb167-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">raw</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">anova</span>(tlc_baseline_1, <span class="at">Terms =</span> <span class="dv">4</span>),</span>
<span id="cb167-8"><a href="#cb167-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">constrain baseline</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">anova</span>(tlc_baseline_2, <span class="at">Terms =</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>),</span>
<span id="cb167-9"><a href="#cb167-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">baseline change</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">anova</span>(tlc_baseline_3, <span class="at">L =</span> L), <span class="co"># same as tlc_baseline_1 as expected</span></span>
<span id="cb167-10"><a href="#cb167-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">ancova</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">anova</span>(tlc_baseline_4, <span class="at">Terms =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">5</span>))) <span class="sc">%&gt;%</span>  <span class="co"># similarly efficient as baseline 2 model</span></span>
<span id="cb167-11"><a href="#cb167-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb167-12"><a href="#cb167-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Chisq</span><span class="st">`</span> <span class="ot">=</span> numDF <span class="sc">*</span> <span class="st">`</span><span class="at">F-value</span><span class="st">`</span>,</span>
<span id="cb167-13"><a href="#cb167-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">.after=</span> <span class="st">`</span><span class="at">F-value</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                   numDF  F-value    Chisq      p-value
raw                    3 35.92934 107.7880 1.557061e-20
constrain baseline     3 37.31422 111.9427 3.074120e-21
baseline change        3 35.92872 107.7862 8.227085e-20
ancova                 3 37.04297 111.1289 2.517144e-20</code></pre>
</div>
</div>
<p>We see that 2 and 4 have larger F-statistics, implying that there is slightly more power to detect the group differences (assuming the assumptions are reasonable).</p>
<p><span class="citation" data-cites="fitzmaurice_applied_2011">Fitzmaurice, Laird, and Ware (<a href="#ref-fitzmaurice_applied_2011" role="doc-biblioref">2011</a>)</span> concludes that 2 should be recommended, because we get similar efficiency gains, and there is an implicit assumption about the covariance matrix of the ancova model. In the ANCOVA model, we only have a 3x3 covariance matrix, where as in baseline change model, we estimated a 4x4 covariance matrix. Hence, we are implicitly assuming that <span class="math inline">\mathrm{Cov}(Y_{i1}, Y_{i2}) = \mathrm{Cov}(Y_{i1}, Y_{i3}) = \mathrm{Cov}(Y_{i1}, Y_{i4})</span></p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-65_d9177e698d94e1df26ef0bd4175d80d4">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a>tlc_baseline_4 <span class="sc">%&gt;%</span> <span class="fu">getVarCov</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Marginal variance covariance matrix
       [,1]   [,2]   [,3]
[1,] 30.151 20.864 12.991
[2,] 20.864 32.230 13.460
[3,] 12.991 13.460 39.478
  Standard Deviations: 5.491 5.6771 6.2831 </code></pre>
</div>
</div>
</section>
<section id="references" class="level2" data-number="21.3">
<h2 data-number="21.3" class="anchored" data-anchor-id="references"><span class="header-section-number">21.3</span> References</h2>
<ul>
<li><a href="https://www.ema.europa.eu/en/documents/scientific-guideline/points-consider-adjustment-baseline-covariates_en.pdf">European Agency for evaluation of medicinal products</a> from a quick google search</li>
</ul>
</section>
</section>
<section id="residuals" class="level1" data-number="22">
<h1 data-number="22"><span class="header-section-number">22</span> Residuals</h1>
<p>Residuals from a longitudinal model will be correlated. <span class="citation" data-cites="fitzmaurice_applied_2011">Fitzmaurice, Laird, and Ware (<a href="#ref-fitzmaurice_applied_2011" role="doc-biblioref">2011</a>)</span> recommends that we “decorrelate” them with a cholesky decomposition of the estimated covariance of the errors. It can get confusing with the estimators and terminology here…</p>
<p><span class="math display">
\begin{aligned}
\varepsilon &amp;= Y_i - X_i\beta \\
\hat\varepsilon &amp;= Y_i - X_i \hat\beta \\
\mathrm{Cov}(\varepsilon) &amp;= \Sigma \\
\widehat{\mathrm{Cov}}(\varepsilon) &amp;= \hat\Sigma \\
\mathrm{Cov}(\hat\varepsilon) &amp;= ??
\end{aligned}
</span></p>
<p>We assume that <span class="math inline">\mathrm{Cov}(\hat \varepsilon) \approx \mathrm{Cov}(\varepsilon)</span>.</p>
<section id="decorrelation" class="level2" data-number="22.1">
<h2 data-number="22.1" class="anchored" data-anchor-id="decorrelation"><span class="header-section-number">22.1</span> Decorrelation</h2>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-66_52bff609b030cb20b15752e4228a5e64">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="co"># can extract residuals with type = "normalized"</span></span>
<span id="cb171-2"><a href="#cb171-2" aria-hidden="true" tabindex="-1"></a>fat_post <span class="sc">%&gt;%</span> <span class="fu">count</span>(id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     id  n
1     1  6
2     2  9
3     3  8
4     4  6
5     5  7
6     6  8
7     7  9
8     8  3
9     9  7
10   10  7
11   11  7
12   12  6
13   13  5
14   14  8
15   15  9
16   16 10
17   17  8
18   18  9
19   19  9
20   20  8
21   21  9
22   22  6
23   23  5
24   24  8
25   25  7
26   26  7
27   27  5
28   28  7
29   29  6
30   30  8
31   31  6
32   32  5
33   33  5
34   34  5
35   35  6
36   36  7
37   37  6
38   38  8
39   39  6
40   40  6
41   41  5
42   42  9
43   43  5
44   44  3
45   45  7
46   46  7
47   47  6
48   48  5
49   49  7
50   50  8
51   51  8
52   52  7
53   53  8
54   54  7
55   55  9
56   56  8
57   57  6
58   58  5
59   59  5
60   60  5
61   61  8
62   62  8
63   63  7
64   64  6
65   65  7
66   66  8
67   67  8
68   68  7
69   69  7
70   70  3
71   71  8
72   72  6
73   73  8
74   74  8
75   75  7
76   76  6
77   77  3
78   78  8
79   79  7
80   80  8
81   81  6
82   82  6
83   83  6
84   84  6
85   85  6
86   86  8
87   87  8
88   88  5
89   89  3
90   90  3
91   91  5
92   92  8
93   93  7
94   94  6
95   95  7
96   96  6
97   97  5
98   98  8
99   99  8
100 100  8
101 101  8
102 102  8
103 103  6
104 104  5
105 105  5
106 106  5
107 107  6
108 108  4
109 109  7
110 110  7
111 111  8
112 112  7
113 113  7
114 114  7
115 115  7
116 116  7
117 117  6
118 118  5
119 119  5
120 120  7
121 121  7
122 122  7
123 123  7
124 124  5
125 125  6
126 126  4
127 127  5
128 128  7
129 129  7
130 130  6
131 131  6
132 132  7
133 133  7
134 134  4
135 135  7
136 136  6
137 137  7
138 138  7
139 139  7
140 140  7
141 141  7
142 142  7
143 143  5
144 144  6
145 145  7
146 146  7
147 147  5
148 148  5
149 149  5
150 150  6
151 151  6
152 152  6
153 153  6
154 154  6
155 155  6
156 156  6
157 157  6
158 158  6
159 159  5
160 160  3
161 161  5
162 162  5</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb173"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a>S <span class="ot">&lt;-</span> <span class="fu">getVarCov</span>(fat_lme, <span class="at">type =</span> <span class="st">"marginal"</span>, <span class="at">individuals =</span> <span class="dv">1</span>)</span>
<span id="cb173-2"><a href="#cb173-2" aria-hidden="true" tabindex="-1"></a>S</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>id 1 
Marginal variance covariance matrix
       1      2      3      4      5      6
1 60.288 46.991 43.546 39.949 36.007 32.886
2 46.991 54.304 42.885 40.853 38.553 35.311
3 43.546 42.885 51.763 41.668 40.846 37.496
4 39.949 40.853 41.668 51.991 43.240 39.776
5 36.007 38.553 40.846 43.240 55.056 42.044
6 32.886 35.311 37.496 39.776 42.044 48.858
  Standard Deviations: 7.7645 7.3691 7.1946 7.2105 7.42 6.9898 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a>R <span class="ot">&lt;-</span> S[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> <span class="fu">chol</span>()</span>
<span id="cb175-2"><a href="#cb175-2" aria-hidden="true" tabindex="-1"></a>R</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>         1        2        3        4        5        6
1 7.764524 6.051983 5.608287 5.145088 4.637412 4.235458
2 0.000000 4.204493 2.127138 2.310672 2.494246 2.301845
3 0.000000 0.000000 3.973052 1.987756 2.399240 2.226406
4 0.000000 0.000000 0.000000 4.028538 2.196216 2.045436
5 0.000000 0.000000 0.000000 0.000000 4.092661 1.668148
6 0.000000 0.000000 0.000000 0.000000 0.000000 3.700942</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb177"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb177-1"><a href="#cb177-1" aria-hidden="true" tabindex="-1"></a><span class="fu">crossprod</span>(R) <span class="co"># R'R</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>         1        2        3        4        5        6
1 60.28784 46.99077 43.54568 39.94916 36.00730 32.88632
2 46.99077 54.30426 42.88479 40.85319 38.55258 35.31101
3 43.54568 42.88479 51.76274 41.66771 40.84585 37.49563
4 39.94916 40.85319 41.66771 51.99143 43.23992 39.77628
5 36.00730 38.55258 40.84585 43.23992 55.05645 42.04400
6 32.88632 35.31101 37.49563 39.77628 42.04400 48.85798</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-67_5c6dfb42619311ccc695d37a59c87eac">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a>eps_hat <span class="ot">&lt;-</span> <span class="fu">residuals</span>(fat_lme, <span class="at">type =</span> <span class="st">"pearson"</span>)</span>
<span id="cb179-2"><a href="#cb179-2" aria-hidden="true" tabindex="-1"></a>eps_hat[fat_post<span class="sc">$</span>id <span class="sc">==</span> <span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>         1          1          1          1          1          1 
-1.4696365  0.6821217 -0.3313941  2.4943915 -2.0320743  0.2480700 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a>Rinv <span class="ot">&lt;-</span> <span class="fu">solve</span>(R)</span>
<span id="cb181-2"><a href="#cb181-2" aria-hidden="true" tabindex="-1"></a>eps_hat[fat_post<span class="sc">$</span>id <span class="sc">==</span> <span class="dv">1</span>] <span class="sc">%*%</span> Rinv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>              1         2           3         4          5          6
[1,] -0.1892758 0.4346816 -0.04895698 0.6357493 -0.8594188 0.07874242</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-68_402e140c7f7f4c45c57d5553d4a9dce7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb183"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a><span class="fu">residuals</span>(fat_lme, <span class="at">type =</span> <span class="st">"normalized"</span>)[fat_post<span class="sc">$</span>id <span class="sc">==</span> <span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>         1          1          1          1          1          1 
-1.4696365  0.6821217 -0.3313941  2.4943915 -2.0320743  0.2480700 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a>fat_lme<span class="sc">$</span>modelStruct</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>reStruct  parameters:
       id1        id2        id3        id4        id5        id6 
 0.7894148 -0.9241612 -1.3828910  0.1210868 -0.2928406 -0.3762166 </code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-69_d977aa715638d384c5fc894f07bd07ac">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a>?nlme<span class="sc">:::</span>recalc</span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a>nlme<span class="sc">:::</span>residuals.lme</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>function (object, level = Q, type = c("response", "pearson", 
    "normalized"), asList = FALSE, ...) 
{
    type &lt;- match.arg(type)
    Q &lt;- object$dims$Q
    val &lt;- object[["residuals"]]
    if (is.character(level)) {
        nlevel &lt;- match(level, names(val))
        if (any(aux &lt;- is.na(nlevel))) {
            stop(sprintf(ngettext(sum(aux), "nonexistent level %s", 
                "nonexistent levels %s"), level[aux]), domain = NA)
        }
        level &lt;- nlevel
    }
    else {
        level &lt;- 1 + level
    }
    if (type != "response") {
        val &lt;- val[, level]/attr(val, "std")
    }
    else {
        val &lt;- val[, level]
    }
    if (type == "normalized") {
        if (!is.null(cSt &lt;- object$modelStruct$corStruct)) {
            val &lt;- recalc(cSt, list(Xy = as.matrix(val)))$Xy[, 
                seq_along(level)]
        }
        else {
            type &lt;- "pearson"
        }
    }
    if (length(level) == 1) {
        grps &lt;- as.character(object[["groups"]][, max(c(1, level - 
            1))])
        if (asList) {
            val &lt;- as.list(split(val, ordered(grps, levels = unique(grps))))
        }
        else {
            grp.nm &lt;- row.names(object[["groups"]])
            val &lt;- naresid(object$na.action, val)
            names(val) &lt;- grps[match(names(val), grp.nm)]
        }
        attr(val, "label") &lt;- switch(type, response = {
            if (!is.null(aux &lt;- attr(object, "units")$y)) paste("Residuals", 
                aux) else "Residuals"
        }, pearson = "Standardized residuals", normalized = "Normalized residuals")
        val
    }
    else naresid(object$na.action, val)
}
&lt;bytecode: 0x7f7ee5f19a40&gt;
&lt;environment: namespace:nlme&gt;</code></pre>
</div>
</div>
</section>
<section id="aggregation" class="level2" data-number="22.2">
<h2 data-number="22.2" class="anchored" data-anchor-id="aggregation"><span class="header-section-number">22.2</span> Aggregation</h2>
<p>The next method deals with aggregation</p>
</section>
</section>
<section id="econometrics-approaches" class="level1" data-number="23">
<h1 data-number="23"><span class="header-section-number">23</span> Econometrics Approaches</h1>
<p>The Econometrics methods that deal with longitudinal data is a little different than statistics. We’ll explore the terminology and methods here.</p>
<ul>
<li>Pooled Model</li>
</ul>
</section>
<section id="software-comparisons" class="level1" data-number="24">
<h1 data-number="24"><span class="header-section-number">24</span> Software Comparisons</h1>
<ul>
<li>plm (unequal observations between groups, needs adjustment from nlme)</li>
<li>lme4, nlme (maximum likelihood approach, not intuitive)</li>
</ul>
</section>
<section id="plm-vignette" class="level1" data-number="25">
<h1 data-number="25"><span class="header-section-number">25</span> PLM Vignette</h1>
<p>Provides functions for panel data from “econometricians” point of view.</p>
<p>Provides a few functions</p>
<section id="plm" class="level2" data-number="25.1">
<h2 data-number="25.1" class="anchored" data-anchor-id="plm"><span class="header-section-number">25.1</span> <code>plm</code></h2>
<ul>
<li>the fixed effects model (“within”),</li>
<li>the pooling model (“pooling”),</li>
<li>the first-difference model (“fd”),</li>
<li>the between model (“between”),</li>
<li>the error components model (“random”).</li>
</ul>
<section id="empluk" class="level3" data-number="25.1.1">
<h3 data-number="25.1.1" class="anchored" data-anchor-id="empluk"><span class="header-section-number">25.1.1</span> EmplUK</h3>
<ul>
<li>firm - firm index</li>
<li>year - year</li>
<li>sector - the sector of activity</li>
<li>emp - employment</li>
<li>wage - wages</li>
<li>capital - capital</li>
<li>output - output</li>
</ul>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-70_69311d5da64bba3b91460b60e3044b52">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"EmplUK"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-71_948e32491af018d6dd297c94899a27d1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb190"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb190-1"><a href="#cb190-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pdata.frame creats a dataframe for plm to work with</span></span>
<span id="cb190-2"><a href="#cb190-2" aria-hidden="true" tabindex="-1"></a>E <span class="ot">&lt;-</span> <span class="fu">pdata.frame</span>(EmplUK,                 <span class="co"># The orig dataframe</span></span>
<span id="cb190-3"><a href="#cb190-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">index=</span><span class="fu">c</span>(<span class="st">"firm"</span>,<span class="st">"year"</span>),      <span class="co"># The "individual" and "time" variable names</span></span>
<span id="cb190-4"><a href="#cb190-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">drop.index =</span> <span class="cn">TRUE</span>,           <span class="co"># Don't show index columns</span></span>
<span id="cb190-5"><a href="#cb190-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">TRUE</span>)            <span class="co"># Show rownames that combine individual-time</span></span>
<span id="cb190-6"><a href="#cb190-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-7"><a href="#cb190-7" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(E) <span class="co"># pdata.frame</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "pdata.frame" "data.frame" </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(E<span class="sc">$</span>emp) <span class="co"># Selecting the column from a pdata.frame give a "pseries" object</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>total sum of squares: 261539.4 
         id        time 
0.980765381 0.009108488 

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.104   1.181   2.287   7.892   7.020 108.562 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="fu">methods</span>(<span class="at">class=</span><span class="st">"pseries"</span>) <span class="co"># Looking at the methods with a pseries object</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] as.matrix         between           Between           Complex          
 [5] D                 diff              F                 G                
 [9] index             is.pbalanced      is.pconsecutive   L                
[13] lag               lead              make.pbalanced    make.pconsecutive
[17] Math              Ops               pcdtest           pdim             
[21] plot              print             pvar              Sum              
[25] summary           Within           
see '?methods' for accessing help and source code</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb196"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb196-1"><a href="#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plm:::print.summary.pseries</span></span>
<span id="cb196-2"><a href="#cb196-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plm:::summary.pseries # The function for summarizing the series</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>pseries datatype comes with various functions to operate on them to compute correctly time lag within each individual</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-72_fb808961760364be5c15fe317ba4c555">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb197"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="fu">between</span>(E<span class="sc">$</span>emp) <span class="co"># means across the time periods</span></span>
<span id="cb197-2"><a href="#cb197-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Sum</span>(E<span class="sc">$</span>emp) <span class="co"># sum across the time periods</span></span>
<span id="cb197-3"><a href="#cb197-3" aria-hidden="true" tabindex="-1"></a><span class="fu">Within</span>(E<span class="sc">$</span>emp) <span class="co"># deviation from mean</span></span>
<span id="cb197-4"><a href="#cb197-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(plm<span class="sc">:::</span><span class="fu">lag.pseries</span>(E<span class="sc">$</span>emp, <span class="dv">0</span><span class="sc">:</span><span class="dv">2</span>)) <span class="co"># Lag the sequence within subject by however many</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="grunfield-example" class="level3" data-number="25.1.2">
<h3 data-number="25.1.2" class="anchored" data-anchor-id="grunfield-example"><span class="header-section-number">25.1.2</span> Grunfield example</h3>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-73_274d74309d681692c3d67fac40e5e28e">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(Grunfeld)</span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a>Grunfeld<span class="sc">$</span>firm <span class="ot">&lt;-</span> <span class="fu">factor</span>(Grunfeld<span class="sc">$</span>firm)</span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a>Grunfeld<span class="sc">$</span>year <span class="ot">&lt;-</span> <span class="fu">factor</span>(Grunfeld<span class="sc">$</span>year)</span>
<span id="cb198-4"><a href="#cb198-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-5"><a href="#cb198-5" aria-hidden="true" tabindex="-1"></a><span class="fu">skim</span>(Grunfeld)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table style="width: auto;" class="table table-condensed">
<caption>
Data summary
</caption>
<tbody>
<tr>
<td style="text-align:left;">
Name
</td>
<td style="text-align:left;">
Grunfeld
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of rows
</td>
<td style="text-align:left;">
200
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of columns
</td>
<td style="text-align:left;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
_______________________
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Column type frequency:
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
factor
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
numeric
</td>
<td style="text-align:left;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
________________________
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
Group variables
</td>
<td style="text-align:left;">
None
</td>
</tr>
</tbody>

</table>
<p><strong>Variable type: factor</strong></p>

<table>
<thead>
<tr>
<th style="text-align:left;">
skim_variable
</th>
<th style="text-align:right;">
n_missing
</th>
<th style="text-align:right;">
complete_rate
</th>
<th style="text-align:left;">
ordered
</th>
<th style="text-align:right;">
n_unique
</th>
<th style="text-align:left;">
top_counts
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
firm
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
1: 20, 2: 20, 3: 20, 4: 20
</td>
</tr>
<tr>
<td style="text-align:left;">
year
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:left;">
193: 10, 193: 10, 193: 10, 193: 10
</td>
</tr>
</tbody>

</table>
<p><strong>Variable type: numeric</strong></p>

<table>
<thead>
<tr>
<th style="text-align:left;">
skim_variable
</th>
<th style="text-align:right;">
n_missing
</th>
<th style="text-align:right;">
complete_rate
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
sd
</th>
<th style="text-align:right;">
p0
</th>
<th style="text-align:right;">
p25
</th>
<th style="text-align:right;">
p50
</th>
<th style="text-align:right;">
p75
</th>
<th style="text-align:right;">
p100
</th>
<th style="text-align:left;">
hist
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
inv
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
145.96
</td>
<td style="text-align:right;">
216.88
</td>
<td style="text-align:right;">
0.93
</td>
<td style="text-align:right;">
33.56
</td>
<td style="text-align:right;">
57.48
</td>
<td style="text-align:right;">
138.04
</td>
<td style="text-align:right;">
1486.7
</td>
<td style="text-align:left;">
▇▁▁▁▁
</td>
</tr>
<tr>
<td style="text-align:left;">
value
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1081.68
</td>
<td style="text-align:right;">
1314.47
</td>
<td style="text-align:right;">
58.12
</td>
<td style="text-align:right;">
199.98
</td>
<td style="text-align:right;">
517.95
</td>
<td style="text-align:right;">
1679.85
</td>
<td style="text-align:right;">
6241.7
</td>
<td style="text-align:left;">
▇▂▁▁▁
</td>
</tr>
<tr>
<td style="text-align:left;">
capital
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
276.02
</td>
<td style="text-align:right;">
301.10
</td>
<td style="text-align:right;">
0.80
</td>
<td style="text-align:right;">
79.17
</td>
<td style="text-align:right;">
205.60
</td>
<td style="text-align:right;">
358.10
</td>
<td style="text-align:right;">
2226.3
</td>
<td style="text-align:left;">
▇▁▁▁▁
</td>
</tr>
</tbody>

</table>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-74_ef8d8eb3b27e9eaafb6f2e32077594a5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a>Grunfeld <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(year, inv, <span class="at">color =</span> <span class="fu">factor</span>(firm))) <span class="sc">+</span> </span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>geom_path: Each group consists of only one observation. Do you need to adjust
the group aesthetic?</code></pre>
</div>
<div class="cell-output-display">
<p><img src="longitudinal_files/figure-html/unnamed-chunk-74-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-75_e7c7630b3ebd42a1380053f998f25317">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a>grun.po <span class="ot">&lt;-</span> <span class="fu">plm</span>(inv<span class="sc">~</span>value <span class="sc">+</span> capital, <span class="at">data =</span> Grunfeld, <span class="at">model =</span> <span class="st">"pooling"</span>) <span class="co"># pooled</span></span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a>grun.fe <span class="ot">&lt;-</span> <span class="fu">plm</span>(inv<span class="sc">~</span>value <span class="sc">+</span> capital, <span class="at">data =</span> Grunfeld, <span class="at">model =</span> <span class="st">"within"</span>) <span class="co"># fixed effects</span></span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a>grun.re <span class="ot">&lt;-</span> <span class="fu">plm</span>(inv<span class="sc">~</span>value <span class="sc">+</span> capital, <span class="at">data =</span> Grunfeld, <span class="at">model =</span> <span class="st">"random"</span>) <span class="co"># random effects</span></span>
<span id="cb201-4"><a href="#cb201-4" aria-hidden="true" tabindex="-1"></a>grun.fd <span class="ot">&lt;-</span> <span class="fu">plm</span>(inv<span class="sc">~</span>value <span class="sc">+</span> capital, <span class="at">data =</span> Grunfeld, <span class="at">model =</span> <span class="st">"fd"</span>) <span class="co"># first difference</span></span>
<span id="cb201-5"><a href="#cb201-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-6"><a href="#cb201-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(grun.po)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Pooling Model

Call:
plm(formula = inv ~ value + capital, data = Grunfeld, model = "pooling")

Balanced Panel: n = 10, T = 20, N = 200

Residuals:
     Min.   1st Qu.    Median   3rd Qu.      Max. 
-291.6757  -30.0137    5.3033   34.8293  369.4464 

Coefficients:
               Estimate  Std. Error t-value  Pr(&gt;|t|)    
(Intercept) -42.7143694   9.5116760 -4.4907 1.207e-05 ***
value         0.1155622   0.0058357 19.8026 &lt; 2.2e-16 ***
capital       0.2306785   0.0254758  9.0548 &lt; 2.2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Total Sum of Squares:    9359900
Residual Sum of Squares: 1755900
R-Squared:      0.81241
Adj. R-Squared: 0.8105
F-statistic: 426.576 on 2 and 197 DF, p-value: &lt; 2.22e-16</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(grun.fe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Oneway (individual) effect Within Model

Call:
plm(formula = inv ~ value + capital, data = Grunfeld, model = "within")

Balanced Panel: n = 10, T = 20, N = 200

Residuals:
      Min.    1st Qu.     Median    3rd Qu.       Max. 
-184.00857  -17.64316    0.56337   19.19222  250.70974 

Coefficients:
        Estimate Std. Error t-value  Pr(&gt;|t|)    
value   0.110124   0.011857  9.2879 &lt; 2.2e-16 ***
capital 0.310065   0.017355 17.8666 &lt; 2.2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Total Sum of Squares:    2244400
Residual Sum of Squares: 523480
R-Squared:      0.76676
Adj. R-Squared: 0.75311
F-statistic: 309.014 on 2 and 188 DF, p-value: &lt; 2.22e-16</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(grun.re)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Oneway (individual) effect Random Effect Model 
   (Swamy-Arora's transformation)

Call:
plm(formula = inv ~ value + capital, data = Grunfeld, model = "random")

Balanced Panel: n = 10, T = 20, N = 200

Effects:
                  var std.dev share
idiosyncratic 2784.46   52.77 0.282
individual    7089.80   84.20 0.718
theta: 0.8612

Residuals:
     Min.   1st Qu.    Median   3rd Qu.      Max. 
-177.6063  -19.7350    4.6851   19.5105  252.8743 

Coefficients:
              Estimate Std. Error z-value Pr(&gt;|z|)    
(Intercept) -57.834415  28.898935 -2.0013  0.04536 *  
value         0.109781   0.010493 10.4627  &lt; 2e-16 ***
capital       0.308113   0.017180 17.9339  &lt; 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Total Sum of Squares:    2381400
Residual Sum of Squares: 548900
R-Squared:      0.7695
Adj. R-Squared: 0.76716
Chisq: 657.674 on 2 DF, p-value: &lt; 2.22e-16</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(grun.fd)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Oneway (individual) effect First-Difference Model

Call:
plm(formula = inv ~ value + capital, data = Grunfeld, model = "fd")

Balanced Panel: n = 10, T = 20, N = 200
Observations used in estimation: 190

Residuals:
       Min.     1st Qu.      Median     3rd Qu.        Max. 
-200.889558  -13.889063    0.016677    9.504223  195.634938 

Coefficients:
              Estimate Std. Error t-value  Pr(&gt;|t|)    
(Intercept) -1.8188902  3.5655931 -0.5101    0.6106    
value        0.0897625  0.0083636 10.7325 &lt; 2.2e-16 ***
capital      0.2917667  0.0537516  5.4281 1.752e-07 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Total Sum of Squares:    584410
Residual Sum of Squares: 345460
R-Squared:      0.40888
Adj. R-Squared: 0.40256
F-statistic: 64.6736 on 2 and 187 DF, p-value: &lt; 2.22e-16</code></pre>
</div>
</div>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-76_30edc51ed731b151dfb01ae96e7d62f1">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Statistical equivalent</span></span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(inv<span class="sc">~</span>value <span class="sc">+</span> capital, <span class="at">data =</span> Grunfeld)) <span class="co"># pooled</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = inv ~ value + capital, data = Grunfeld)

Residuals:
    Min      1Q  Median      3Q     Max 
-291.68  -30.01    5.30   34.83  369.45 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -42.714369   9.511676  -4.491 1.21e-05 ***
value         0.115562   0.005836  19.803  &lt; 2e-16 ***
capital       0.230678   0.025476   9.055  &lt; 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 94.41 on 197 degrees of freedom
Multiple R-squared:  0.8124,    Adjusted R-squared:  0.8105 
F-statistic: 426.6 on 2 and 197 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(inv<span class="sc">~</span>firm <span class="sc">+</span> capital <span class="sc">+</span> value, <span class="at">data =</span> Grunfeld)) <span class="co"># fixed effects</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = inv ~ firm + capital + value, data = Grunfeld)

Residuals:
     Min       1Q   Median       3Q      Max 
-184.009  -17.643    0.563   19.192  250.710 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  -70.29672   49.70796  -1.414    0.159    
firm2        172.20253   31.16126   5.526 1.08e-07 ***
firm3       -165.27512   31.77556  -5.201 5.14e-07 ***
firm4         42.48742   43.90988   0.968    0.334    
firm5        -44.32010   50.49226  -0.878    0.381    
firm6         47.13542   46.81068   1.007    0.315    
firm7          3.74324   50.56493   0.074    0.941    
firm8         12.75106   44.05263   0.289    0.773    
firm9        -16.92555   48.45327  -0.349    0.727    
firm10        63.72887   50.33023   1.266    0.207    
capital        0.31007    0.01735  17.867  &lt; 2e-16 ***
value          0.11012    0.01186   9.288  &lt; 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 52.77 on 188 degrees of freedom
Multiple R-squared:  0.9441,    Adjusted R-squared:  0.9408 
F-statistic: 288.5 on 11 and 188 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lmer</span>(inv<span class="sc">~</span> (<span class="dv">1</span><span class="sc">|</span>firm) <span class="sc">+</span> capital <span class="sc">+</span> value, <span class="at">data =</span> Grunfeld)) <span class="co"># random model, but different random effect estimators</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Some predictor variables are on very different scales: consider
rescaling</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML ['lmerMod']
Formula: inv ~ (1 | firm) + capital + value
   Data: Grunfeld

REML criterion at convergence: 2195.9

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-3.4319 -0.3498  0.0210  0.3592  4.8145 

Random effects:
 Groups   Name        Variance Std.Dev.
 firm     (Intercept) 7367     85.83   
 Residual             2781     52.74   
Number of obs: 200, groups:  firm, 10

Fixed effects:
             Estimate Std. Error t value
(Intercept) -57.86442   29.37776   -1.97
capital       0.30819    0.01717   17.95
value         0.10979    0.01053   10.43

Correlation of Fixed Effects:
        (Intr) capitl
capital -0.019       
value   -0.328 -0.368
fit warnings:
Some predictor variables are on very different scales: consider rescaling</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="gls" class="level1" data-number="26">
<h1 data-number="26"><span class="header-section-number">26</span> GLS</h1>
<p>See <a href="https://en.wikipedia.org/wiki/Generalized_least_squares">Wikipedia</a>, the article is fairly comprehensive.</p>
<p>Common forms of the <code>gls</code> command are as follows (from <span class="citation" data-cites="pinheiro_mixed-effects_2000">Pinheiro, Pinheiro, and Bates (<a href="#ref-pinheiro_mixed-effects_2000" role="doc-biblioref">2000</a>)</span>):</p>
<ul>
<li><code>gls( model, data, correlation )</code> - correlated errors</li>
<li><code>gls( model, data, weights )</code> - heteroscedastic errors</li>
<li><code>gls( model, data, correlation, weights )</code> - both</li>
</ul>
<p>We can play around with the structure of covariance matrices with:</p>
<div class="cell" data-hash="longitudinal_cache/html/unnamed-chunk-78_62f02c5ded49dc2966c0474b4ec48d30">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="co"># In order to specify the correlation structure to be unstructured, we allow each group to be different for id. But for each time point to be related.</span></span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ?corSymm</span></span>
<span id="cb216-3"><a href="#cb216-3" aria-hidden="true" tabindex="-1"></a><span class="co"># corSymm(form=tlc$time | tlc$id)</span></span>
<span id="cb216-4"><a href="#cb216-4" aria-hidden="true" tabindex="-1"></a><span class="co"># In order to see all the correlation matrices for just the first individual</span></span>
<span id="cb216-5"><a href="#cb216-5" aria-hidden="true" tabindex="-1"></a><span class="fu">corMatrix</span>(<span class="fu">Initialize</span>(<span class="fu">corSymm</span>(<span class="at">form=</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> id), <span class="at">data=</span>tlc))<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>              [,1]          [,2]          [,3]          [,4]
[1,]  1.000000e+00 6.925961e-310 6.925961e-310 6.925961e-310
[2,] 6.925961e-310  1.000000e+00 6.925961e-310 6.925961e-310
[3,] 6.925961e-310 6.925961e-310  1.000000e+00 6.925961e-310
[4,] 6.925961e-310 6.925961e-310 6.925961e-310  1.000000e+00</code></pre>
</div>
</div>
</section>
<section id="References" class="level1" data-number="27">
<h1 data-number="27"><span class="header-section-number">27</span> References</h1>


<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-fitzmaurice_applied_2011" class="csl-entry" role="doc-biblioentry">
Fitzmaurice, Garrett M, Nan M Laird, and James H Ware. 2011. <em>Applied Longitudinal Analysis</em>. John Wiley &amp; Sons. <a href="https://content.sph.harvard.edu/fitzmaur/ala2e/">https://content.sph.harvard.edu/fitzmaur/ala2e/</a>.
</div>
<div id="ref-pinheiro_mixed-effects_2000" class="csl-entry" role="doc-biblioentry">
Pinheiro, José, Jose ́ C. Pinheiro, and Douglas Bates. 2000. <em>Mixed-Effects Models in s and s-<span>PLUS</span></em>. Springer Science &amp; Business Media.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../gam/gam.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Generalized Additive Models</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../longitudinal/randomcoef.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Random Coefficient Models (CALS)</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>